{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load Library"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import DataFrame\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from Bio import Entrez \n",
    "import pycurl\n",
    "import matplotlib.pyplot as plt\n",
    "import os, sys\n",
    "from tqdm import tqdm_notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_GenomeAssembly(G, S):\n",
    "    F = G+'_'+S+'.csv'\n",
    "    if not os.path.isfile(F):\n",
    "        with open(F, 'wb') as f:\n",
    "            c = pycurl.Curl()\n",
    "            c.setopt(c.URL, 'https://www.ncbi.nlm.nih.gov/genomes/Genome2BE/genome2srv.cgi?action=download&orgn=%22'\n",
    "                     +G+'%20'+S+\n",
    "                     '%22[orgn]&status=50|40||30|20&report=proks&group=--%20All%20Prokaryotes%20--&subgroup=--%20All%20Prokaryotes%20--&format=csv')\n",
    "            c.setopt(c.WRITEDATA, f)\n",
    "            c.perform()\n",
    "            c.close()\n",
    "    return F"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load Dataset to Scrap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0               Propionibacterium acnes\n",
       "1            Staphylococcus epidermidis\n",
       "2    Corynebacterium tuberculostearicum\n",
       "3                Staphylococcus capitis\n",
       "4              Corynebacterium simulans\n",
       "5                   Streptococcus mitis\n",
       "6                Staphylococcus hominis\n",
       "7           Corynebacterium aurimucosum\n",
       "8        Corynebacterium kroppenstedtii\n",
       "9            Corynebacterium amycolatum\n",
       "Name: Sebaceous, dtype: object"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Pilih data dari Sebaceous\n",
    "df_Byrd = pd.read_csv(\"Byrd_2018_Bacteria.csv\")\n",
    "df_Sebaceous = df_Byrd.loc[:,'Sebaceous']\n",
    "df_Sebaceous"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Iteration Test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f817ac429b2348cb91b377470e6cfcae",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(IntProgress(value=0, max=10), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "#Email for NCBI\n",
    "Entrez.email = \"matin_nuhamunada@mail.ugm.ac.id\"\n",
    "\n",
    "#Pilih data dari Sebaceous\n",
    "df_Byrd = pd.read_csv(\"Byrd_2018_Bacteria.csv\")\n",
    "df_Sebaceous = df_Byrd.loc[:,'Sebaceous']\n",
    "df_Sebaceous\n",
    "\n",
    "#Add information on biome\n",
    "df_NCBI = DataFrame(columns=(\"IdList\",\"QueryTranslation\",\"Genus\",\"Species\",\"NoStrain\",\"Size\",\"GC\",\"Genes\",\"Proteins\"))\n",
    "#df2.index.name = 'No'\n",
    "\n",
    "for i in tqdm_notebook(range(len(df_Sebaceous))):\n",
    "    handle = Entrez.esearch(db=\"genome\", term=df_Sebaceous[i])\n",
    "    record = Entrez.read(handle)\n",
    "    df_NCBI.loc[i, \"IdList\"] = record['IdList']\n",
    "    df_NCBI.loc[i, \"QueryTranslation\"] = record['QueryTranslation']\n",
    "    string = df_NCBI.loc[i, \"QueryTranslation\"]\n",
    "    string = string.replace('[Organism]', '')\n",
    "    string = string.replace('\"', '')\n",
    "    string = string.split()\n",
    "    df_NCBI.loc[i, \"Genus\"] = string[0]\n",
    "    df_NCBI.loc[i, \"Species\"] = string[1]\n",
    "    #print(record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "34cc2b68492a474db1e4a522d3384b30",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(IntProgress(value=0, max=10), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "for i in tqdm_notebook(range(len(df_NCBI))):\n",
    "    get_GenomeAssembly(df_NCBI.loc[i, \"Genus\"], df_NCBI.loc[i, \"Species\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "ad294324c10f4e2c86d16286fdbd76e5",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(IntProgress(value=0, max=10), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IdList</th>\n",
       "      <th>QueryTranslation</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>NoStrain</th>\n",
       "      <th>Size</th>\n",
       "      <th>GC</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Proteins</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[1140]</td>\n",
       "      <td>\"Cutibacterium acnes\"[Organism]</td>\n",
       "      <td>Cutibacterium</td>\n",
       "      <td>acnes</td>\n",
       "      <td>189</td>\n",
       "      <td>2.51243</td>\n",
       "      <td>59.9714</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[155]</td>\n",
       "      <td>\"Staphylococcus epidermidis\"[Organism] OR Stap...</td>\n",
       "      <td>Staphylococcus</td>\n",
       "      <td>epidermidis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[2024]</td>\n",
       "      <td>\"Corynebacterium tuberculostearicum\"[Organism]</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>tuberculostearicum</td>\n",
       "      <td>1</td>\n",
       "      <td>2.37226</td>\n",
       "      <td>60</td>\n",
       "      <td>2271</td>\n",
       "      <td>2171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[2054]</td>\n",
       "      <td>\"Staphylococcus capitis\"[Organism]</td>\n",
       "      <td>Staphylococcus</td>\n",
       "      <td>capitis</td>\n",
       "      <td>68</td>\n",
       "      <td>2.47772</td>\n",
       "      <td>32.8599</td>\n",
       "      <td>2408.74</td>\n",
       "      <td>2288.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[43748]</td>\n",
       "      <td>\"Corynebacterium simulans\"[Organism]</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>simulans</td>\n",
       "      <td>3</td>\n",
       "      <td>2.6633</td>\n",
       "      <td>59.1667</td>\n",
       "      <td>2510</td>\n",
       "      <td>2358.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>[530]</td>\n",
       "      <td>\"Streptococcus mitis\"[Organism]</td>\n",
       "      <td>Streptococcus</td>\n",
       "      <td>mitis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>[2014]</td>\n",
       "      <td>\"Staphylococcus hominis\"[Organism]</td>\n",
       "      <td>Staphylococcus</td>\n",
       "      <td>hominis</td>\n",
       "      <td>65</td>\n",
       "      <td>2.25821</td>\n",
       "      <td>31.4626</td>\n",
       "      <td>2252.98</td>\n",
       "      <td>2130.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>[2077]</td>\n",
       "      <td>\"Corynebacterium aurimucosum\"[Organism]</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>aurimucosum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>[2146]</td>\n",
       "      <td>\"Corynebacterium kroppenstedtii\"[Organism]</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>kroppenstedtii</td>\n",
       "      <td>6</td>\n",
       "      <td>2.5158</td>\n",
       "      <td>56.95</td>\n",
       "      <td>2163.67</td>\n",
       "      <td>2042.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>[2023]</td>\n",
       "      <td>\"Corynebacterium amycolatum\"[Organism]</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>amycolatum</td>\n",
       "      <td>6</td>\n",
       "      <td>2.5245</td>\n",
       "      <td>58.8</td>\n",
       "      <td>2218.17</td>\n",
       "      <td>2104.17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    IdList                                   QueryTranslation  \\\n",
       "0   [1140]                    \"Cutibacterium acnes\"[Organism]   \n",
       "1    [155]  \"Staphylococcus epidermidis\"[Organism] OR Stap...   \n",
       "2   [2024]     \"Corynebacterium tuberculostearicum\"[Organism]   \n",
       "3   [2054]                 \"Staphylococcus capitis\"[Organism]   \n",
       "4  [43748]               \"Corynebacterium simulans\"[Organism]   \n",
       "5    [530]                    \"Streptococcus mitis\"[Organism]   \n",
       "6   [2014]                 \"Staphylococcus hominis\"[Organism]   \n",
       "7   [2077]            \"Corynebacterium aurimucosum\"[Organism]   \n",
       "8   [2146]         \"Corynebacterium kroppenstedtii\"[Organism]   \n",
       "9   [2023]             \"Corynebacterium amycolatum\"[Organism]   \n",
       "\n",
       "             Genus             Species NoStrain     Size       GC    Genes  \\\n",
       "0    Cutibacterium               acnes      189  2.51243  59.9714      NaN   \n",
       "1   Staphylococcus         epidermidis      NaN      NaN      NaN      NaN   \n",
       "2  Corynebacterium  tuberculostearicum        1  2.37226       60     2271   \n",
       "3   Staphylococcus             capitis       68  2.47772  32.8599  2408.74   \n",
       "4  Corynebacterium            simulans        3   2.6633  59.1667     2510   \n",
       "5    Streptococcus               mitis      NaN      NaN      NaN      NaN   \n",
       "6   Staphylococcus             hominis       65  2.25821  31.4626  2252.98   \n",
       "7  Corynebacterium         aurimucosum      NaN      NaN      NaN      NaN   \n",
       "8  Corynebacterium      kroppenstedtii        6   2.5158    56.95  2163.67   \n",
       "9  Corynebacterium          amycolatum        6   2.5245     58.8  2218.17   \n",
       "\n",
       "  Proteins  \n",
       "0      NaN  \n",
       "1      NaN  \n",
       "2     2171  \n",
       "3  2288.41  \n",
       "4  2358.33  \n",
       "5      NaN  \n",
       "6  2130.86  \n",
       "7      NaN  \n",
       "8  2042.67  \n",
       "9  2104.17  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for i in tqdm_notebook(range(len(df_NCBI))):\n",
    "    df = pd.read_csv(df_NCBI.loc[i, 'Genus']+'_'+df_NCBI.loc[i, 'Species']+'.csv')\n",
    "    df_Complete = df\n",
    "    #df_Complete = df[df['Level'].str.contains(\"Complete Genome|chromosome\")]\n",
    "    df_Complete = df_Complete[['#Organism/Name','Strain', 'Size (Mb)', 'GC%', 'Replicons', 'Level', 'Genes', 'Proteins']]\n",
    "    NoStrain = df_Complete[\"Strain\"].count()\n",
    "    Size = df_Complete[\"Size (Mb)\"].mean()\n",
    "    GC = df_Complete[\"GC%\"].mean()\n",
    "\n",
    "    try:\n",
    "        Genes = df_Complete[\"Genes\"].apply(int)\n",
    "        Genes = Genes.mean()\n",
    "        Proteins = df_Complete[\"Proteins\"].apply(int)  \n",
    "        Proteins = Proteins.mean()\n",
    "        df_NCBI.loc[i, \"NoStrain\"] = NoStrain\n",
    "        df_NCBI.loc[i, \"Size\"] = Size\n",
    "        df_NCBI.loc[i, \"GC\"] = GC\n",
    "        df_NCBI.loc[i, \"Genes\"] = Genes\n",
    "        df_NCBI.loc[i, \"Proteins\"] = Proteins\n",
    "    except ValueError:\n",
    "        pass      # or whatever\n",
    "    \n",
    "df_NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d79ee713ded248e7a3ad42bf5c99d789",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(IntProgress(value=0, max=10), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IdList</th>\n",
       "      <th>QueryTranslation</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>NoStrain</th>\n",
       "      <th>Size</th>\n",
       "      <th>GC</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Proteins</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[1140]</td>\n",
       "      <td>\"Cutibacterium acnes\"[Organism]</td>\n",
       "      <td>Cutibacterium</td>\n",
       "      <td>acnes</td>\n",
       "      <td>18</td>\n",
       "      <td>2.52176</td>\n",
       "      <td>60.0389</td>\n",
       "      <td>2537</td>\n",
       "      <td>2381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[155]</td>\n",
       "      <td>\"Staphylococcus epidermidis\"[Organism] OR Stap...</td>\n",
       "      <td>Staphylococcus</td>\n",
       "      <td>epidermidis</td>\n",
       "      <td>12</td>\n",
       "      <td>2.59849</td>\n",
       "      <td>32.0975</td>\n",
       "      <td>2580.83</td>\n",
       "      <td>2414.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[2024]</td>\n",
       "      <td>\"Corynebacterium tuberculostearicum\"[Organism]</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>tuberculostearicum</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[2054]</td>\n",
       "      <td>\"Staphylococcus capitis\"[Organism]</td>\n",
       "      <td>Staphylococcus</td>\n",
       "      <td>capitis</td>\n",
       "      <td>3</td>\n",
       "      <td>2.49173</td>\n",
       "      <td>32.991</td>\n",
       "      <td>2425</td>\n",
       "      <td>2289.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[43748]</td>\n",
       "      <td>\"Corynebacterium simulans\"[Organism]</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>simulans</td>\n",
       "      <td>2</td>\n",
       "      <td>2.66833</td>\n",
       "      <td>59.15</td>\n",
       "      <td>2503.5</td>\n",
       "      <td>2351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>[530]</td>\n",
       "      <td>\"Streptococcus mitis\"[Organism]</td>\n",
       "      <td>Streptococcus</td>\n",
       "      <td>mitis</td>\n",
       "      <td>3</td>\n",
       "      <td>2.07362</td>\n",
       "      <td>40.0667</td>\n",
       "      <td>2047.67</td>\n",
       "      <td>1903.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>[2014]</td>\n",
       "      <td>\"Staphylococcus hominis\"[Organism]</td>\n",
       "      <td>Staphylococcus</td>\n",
       "      <td>hominis</td>\n",
       "      <td>1</td>\n",
       "      <td>2.25341</td>\n",
       "      <td>31.4</td>\n",
       "      <td>2258</td>\n",
       "      <td>2070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>[2077]</td>\n",
       "      <td>\"Corynebacterium aurimucosum\"[Organism]</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>aurimucosum</td>\n",
       "      <td>1</td>\n",
       "      <td>2.81923</td>\n",
       "      <td>60.5248</td>\n",
       "      <td>2663</td>\n",
       "      <td>2525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>[2146]</td>\n",
       "      <td>\"Corynebacterium kroppenstedtii\"[Organism]</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>kroppenstedtii</td>\n",
       "      <td>1</td>\n",
       "      <td>2.4468</td>\n",
       "      <td>57.5</td>\n",
       "      <td>2088</td>\n",
       "      <td>1999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>[2023]</td>\n",
       "      <td>\"Corynebacterium amycolatum\"[Organism]</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>amycolatum</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    IdList                                   QueryTranslation  \\\n",
       "0   [1140]                    \"Cutibacterium acnes\"[Organism]   \n",
       "1    [155]  \"Staphylococcus epidermidis\"[Organism] OR Stap...   \n",
       "2   [2024]     \"Corynebacterium tuberculostearicum\"[Organism]   \n",
       "3   [2054]                 \"Staphylococcus capitis\"[Organism]   \n",
       "4  [43748]               \"Corynebacterium simulans\"[Organism]   \n",
       "5    [530]                    \"Streptococcus mitis\"[Organism]   \n",
       "6   [2014]                 \"Staphylococcus hominis\"[Organism]   \n",
       "7   [2077]            \"Corynebacterium aurimucosum\"[Organism]   \n",
       "8   [2146]         \"Corynebacterium kroppenstedtii\"[Organism]   \n",
       "9   [2023]             \"Corynebacterium amycolatum\"[Organism]   \n",
       "\n",
       "             Genus             Species NoStrain     Size       GC    Genes  \\\n",
       "0    Cutibacterium               acnes       18  2.52176  60.0389     2537   \n",
       "1   Staphylococcus         epidermidis       12  2.59849  32.0975  2580.83   \n",
       "2  Corynebacterium  tuberculostearicum        0      NaN      NaN      NaN   \n",
       "3   Staphylococcus             capitis        3  2.49173   32.991     2425   \n",
       "4  Corynebacterium            simulans        2  2.66833    59.15   2503.5   \n",
       "5    Streptococcus               mitis        3  2.07362  40.0667  2047.67   \n",
       "6   Staphylococcus             hominis        1  2.25341     31.4     2258   \n",
       "7  Corynebacterium         aurimucosum        1  2.81923  60.5248     2663   \n",
       "8  Corynebacterium      kroppenstedtii        1   2.4468     57.5     2088   \n",
       "9  Corynebacterium          amycolatum        0      NaN      NaN      NaN   \n",
       "\n",
       "  Proteins  \n",
       "0     2381  \n",
       "1  2414.75  \n",
       "2      NaN  \n",
       "3  2289.67  \n",
       "4     2351  \n",
       "5  1903.67  \n",
       "6     2070  \n",
       "7     2525  \n",
       "8     1999  \n",
       "9      NaN  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for i in tqdm_notebook(range(len(df_NCBI))):\n",
    "    df = pd.read_csv(df_NCBI.loc[i, 'Genus']+'_'+df_NCBI.loc[i, 'Species']+'.csv')\n",
    "    #df_Complete = df\n",
    "    df_Complete = df[df['Level'].str.contains(\"Complete Genome\")]\n",
    "    df_Complete = df_Complete[['#Organism/Name','Strain', 'Size (Mb)', 'GC%', 'Replicons', 'Level', 'Genes', 'Proteins']]\n",
    "    NoStrain = df_Complete[\"Strain\"].count()\n",
    "    Size = df_Complete[\"Size (Mb)\"].mean()\n",
    "    GC = df_Complete[\"GC%\"].mean()\n",
    "\n",
    "    try:\n",
    "        Genes = df_Complete[\"Genes\"].apply(int)\n",
    "        Genes = Genes.mean()\n",
    "        Proteins = df_Complete[\"Proteins\"].apply(int)  \n",
    "        Proteins = Proteins.mean()\n",
    "    except ValueError:\n",
    "        pass      # or whatever\n",
    "\n",
    "    df_NCBI.loc[i, \"NoStrain\"] = NoStrain\n",
    "    df_NCBI.loc[i, \"Size\"] = Size\n",
    "    df_NCBI.loc[i, \"GC\"] = GC\n",
    "    df_NCBI.loc[i, \"Genes\"] = Genes\n",
    "    df_NCBI.loc[i, \"Proteins\"] = Proteins\n",
    "df_NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "c422ad97dd5145ba89341e82cfa3f0ca",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(IntProgress(value=0, max=10), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "for a in tqdm_notebook(range(len(df_NCBI))):\n",
    "    df = pd.read_csv(df_NCBI.loc[a, 'Genus']+'_'+df_NCBI.loc[a, 'Species']+'.csv')\n",
    "    df_Complete = df[df['Level'].str.contains(\"Complete Genome\")]\n",
    "    df_Complete = df_Complete.reset_index()\n",
    "    \n",
    "    df_Refseq = DataFrame(columns=(\"Organism/Name\",\"Strain\",\"Refseq\"))   \n",
    "    \n",
    "    for i in range(len(df_Complete)):\n",
    "        try:\n",
    "            df_Refseq.loc[i, \"Organism/Name\"] = df_Complete.loc[i, \"#Organism/Name\"]\n",
    "            df_Refseq.loc[i, \"Strain\"] = df_Complete.loc[i, \"Strain\"]\n",
    "            string = df_Complete.loc[i, \"Replicons\"]\n",
    "            string = string.replace('chromosome:', '')\n",
    "            string = string.replace('chromosome 1:', '')\n",
    "            string = string.replace('chromosome I:', '')\n",
    "            string = string.replace('/', ' ')\n",
    "            string = string.split()\n",
    "            df_Refseq.loc[i, \"Refseq\"] = string[0]\n",
    "            df_Refseq.to_csv('Refseq_'+df_NCBI.loc[a, 'Genus']+'_'+df_NCBI.loc[a, 'Species']+'.csv')\n",
    "        except:\n",
    "            print('error '+df_NCBI.loc[a, 'Genus']+'_'+df_NCBI.loc[a, 'Species'])\n",
    "            pass           "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Organism/Name</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Refseq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>Staphylococcus epidermidis ATCC 12228</td>\n",
       "      <td>ATCC 12228</td>\n",
       "      <td>NC_004461.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Staphylococcus epidermidis RP62A</td>\n",
       "      <td>RP62A</td>\n",
       "      <td>NC_002976.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>Staphylococcus epidermidis PM221</td>\n",
       "      <td>PM221</td>\n",
       "      <td>NZ_HG813242.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>SEI</td>\n",
       "      <td>NZ_CP009046.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>14.1.R1</td>\n",
       "      <td>CP018841.1;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>1457</td>\n",
       "      <td>NZ_CP020463.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>ATCC 12228</td>\n",
       "      <td>NZ_CP022247.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>DAR1907</td>\n",
       "      <td>NZ_CP013943.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>FDAARGOS_153</td>\n",
       "      <td>NZ_CP014119.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>FDAARGOS_161</td>\n",
       "      <td>NZ_CP014132.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>CSF41498</td>\n",
       "      <td>NZ_CP030246.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>11</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>BPH0662</td>\n",
       "      <td>NZ_LT571449.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0                          Organism/Name        Strain  \\\n",
       "0            0  Staphylococcus epidermidis ATCC 12228    ATCC 12228   \n",
       "1            1       Staphylococcus epidermidis RP62A         RP62A   \n",
       "2            2       Staphylococcus epidermidis PM221         PM221   \n",
       "3            3             Staphylococcus epidermidis           SEI   \n",
       "4            4             Staphylococcus epidermidis       14.1.R1   \n",
       "5            5             Staphylococcus epidermidis          1457   \n",
       "6            6             Staphylococcus epidermidis    ATCC 12228   \n",
       "7            7             Staphylococcus epidermidis       DAR1907   \n",
       "8            8             Staphylococcus epidermidis  FDAARGOS_153   \n",
       "9            9             Staphylococcus epidermidis  FDAARGOS_161   \n",
       "10          10             Staphylococcus epidermidis      CSF41498   \n",
       "11          11             Staphylococcus epidermidis       BPH0662   \n",
       "\n",
       "           Refseq  \n",
       "0     NC_004461.1  \n",
       "1     NC_002976.3  \n",
       "2   NZ_HG813242.1  \n",
       "3   NZ_CP009046.1  \n",
       "4     CP018841.1;  \n",
       "5   NZ_CP020463.1  \n",
       "6   NZ_CP022247.1  \n",
       "7   NZ_CP013943.1  \n",
       "8   NZ_CP014119.1  \n",
       "9   NZ_CP014132.1  \n",
       "10  NZ_CP030246.1  \n",
       "11  NZ_LT571449.1  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = 1\n",
    "pd.read_csv(\"Refseq_\"+df_NCBI.loc[a, 'Genus']+'_'+df_NCBI.loc[a, 'Species']+'.csv')\n",
    "#x = pd.read_csv(df_NCBI.loc[a, 'Genus']+'_'+df_NCBI.loc[a, 'Species']+'.csv')\n",
    "#x[\"Replicons\"][4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
