{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Mining of Human Skin Microbiome from EBI-Metagenomics Portal\n",
    "\n",
    "_Matin Nuhamunada_<sup>1*</sup>, _Gregorius Altius Pratama_<sup>1</sup>, _Setianing Wikanthi_<sup>2</sup>, and _Mohamad Khoirul Anam_<sup>1</sup>\n",
    "\n",
    "<sup>1</sup>Department of Tropical Biology, Universitas Gadjah Mada;   \n",
    "Jl. Teknika Selatan, Sekip Utara, Bulaksumur, Yogyakarta, Indonesia, 55281;   \n",
    "\n",
    "<sup>2</sup>Department of Agricultural Microbiology, Universitas Gadjah Mada;  \n",
    "\n",
    "*Correspondence: [matin_nuhamunada@ugm.ac.id](mailto:matin_nuhamunada@mail.ugm.ac.id)  \n",
    "[mohamad.khoirul.anam@mail.ugm.ac.id](mailto:mohamad.khoirul.anam@mail.ugm.ac.id)  \n",
    "[gregorius.altius.p@mail.ugm.ac.id](mailto:gregorius.altius.p@mail.ugm.ac.id)  \n",
    "[setianingwikanthi@mail.ugm.ac.id](mailto:setianingwikanthi@mail.ugm.ac.id)\n",
    "\n",
    "---\n",
    "## Abstract\n",
    "Human skin microbiome is unique to individuals in regards to many aspects, including behaviour, environment, and perhaps maybe genes. To understand more about the distribution of human skin microbiome across the globe, we compare several skin microbiome study available in the EBI-Metagenomic Portal. Study data was acquired using EBI-Metagenome API, and sample data was selected based on sex, location, and bodysite. The biological observation matrix from the analysis result of the selected samples were compared using MEGAN. \n",
    "\n",
    "### Keywords\n",
    "Human Skin, Microbiome, EBI-Metagenome\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import Python Modules\n",
    "We use python 3 script using ``pandas``, ``jsonapi_client``, ``pycurl``, to mine the data from EBI-metagenomic portal  [[1]](#ref1)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import DataFrame\n",
    "import pandas as pd\n",
    "\n",
    "try:\n",
    "    from urllib import urlencode\n",
    "except ImportError:\n",
    "    from urllib.parse import urlencode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from jsonapi_client import Session, Filter\n",
    "\n",
    "API_BASE = 'https://www.ebi.ac.uk/metagenomics/api/latest/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pycurl\n",
    "import os, sys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get Study\n",
    "We search the EBI Metagenomic database on human skin microbiome study in the host-associated biome with 'skin' as search term. The study list can be found on this link: https://www.ebi.ac.uk/metagenomics/projects/doExportDetails?searchTerm=skin&includingChildren=true&biomeLineage=root%3AHost-associated%3AHuman&search=Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "data.csv\n"
     ]
    }
   ],
   "source": [
    "filename = 'data.csv'\n",
    "print(filename)\n",
    "if not os.path.isfile(filename):\n",
    "    with open(filename, 'wb') as f:\n",
    "        c = pycurl.Curl()\n",
    "        c.setopt(c.URL, 'https://www.ebi.ac.uk/metagenomics/projects/doExportDetails?searchTerm=skin&includingChildren=true&biomeLineage=root%3AHost-associated%3AHuman&search=Search')\n",
    "        c.setopt(c.WRITEDATA, f)\n",
    "        c.perform()\n",
    "        c.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     Study ID                                         Study Name  \\\n",
      "0   ERP104068  EMG produced TPA metagenomics assembly of the ...   \n",
      "1   SRP002480  Gene-Environment Interactions at the Skin Surface   \n",
      "2   ERP018577  Human skin bacterial and fungal microbiotas an...   \n",
      "3   ERP022958  Impact of the Mk VI SkinSuit on skin microbiot...   \n",
      "4   ERP019566  Longitudinal study of the diabetic skin and wo...   \n",
      "5   ERP016629  Microbiome samples derived from Buruli ulcer w...   \n",
      "6   ERP021525  Micromes on salmon skin and surrounding sea water   \n",
      "7   SRP056364  Skin microbiome in human volunteers inoculated...   \n",
      "8   ERP104518                  skin microbiota in infected frogs   \n",
      "9   ERP104520                 Skin microbiota of Scinax alcatraz   \n",
      "10  ERP104516  Variations on the diversity of amphibian skin ...   \n",
      "\n",
      "    Number Of Samples Submitted Date  Analysis NCBI Project ID  \\\n",
      "0                  45     2017-11-15  Finished      PRJEB22388   \n",
      "1                2560     2016-02-03  Finished      PRJNA46333   \n",
      "2                  96     2016-11-03  Finished      PRJEB16723   \n",
      "3                 204     2017-06-16  Finished      PRJEB20777   \n",
      "4                 258     2017-11-27  Finished      PRJEB17696   \n",
      "5                  14     2016-07-29  Finished      PRJEB14948   \n",
      "6                   5     2017-02-28  Finished      PRJEB19482   \n",
      "7                 191     2016-02-04  Finished     PRJNA278873   \n",
      "8                  21     2017-10-02  Finished      PRJEB22812   \n",
      "9                  13     2017-10-02  Finished      PRJEB22814   \n",
      "10                 32     2017-09-30  Finished      PRJEB22810   \n",
      "\n",
      "   Public Release Date                     Centre Name Experimental Factor  \\\n",
      "0                  NaN                             EMG                null   \n",
      "1                  NaN                           NIAMS                null   \n",
      "2           2016-12-23     Universidade Federal do ABC                null   \n",
      "3                  NaN                           LSHTM                null   \n",
      "4                  NaN            The ithree institute                null   \n",
      "5           2016-10-30  Institute of tropical medicine                null   \n",
      "6                  NaN                          NOFIMA                null   \n",
      "7                  NaN                      BioProject           Infection   \n",
      "8                  NaN       UNIVERSIDADE DE SAO PAULO                null   \n",
      "9                  NaN         University of Sao Paulo                null   \n",
      "10                 NaN         UNIVERSITY OF SAO PAULO                null   \n",
      "\n",
      "   Is Public                                     Study Abstract  \n",
      "0       True  The human gut metagenome Third Party Annotatio...  \n",
      "1       True  16S rRNA gene sequences amplified from subject...  \n",
      "2       True  Using high-throughput 16S rDNA and ITS1 sequen...  \n",
      "3       True  Microgravity induces physiological decondition...  \n",
      "4       True  Background: Type II diabetes is a chronic heal...  \n",
      "5       True  Background: Buruli ulcer (BU) is an infectious...  \n",
      "6       True  Pilot study to investigate if sampled Atlantic...  \n",
      "7       True  The aim of this project was to investigate the...  \n",
      "8       True                  skin microbiota in infected frogs  \n",
      "9       True                 Skin microbiota of Scinax alcatraz  \n",
      "10      True  Variations on the diversity of amphibian skin ...  \n"
     ]
    }
   ],
   "source": [
    "#Script ini untuk mengekstrak data tabel dari file CSV\n",
    "df1 = pd.read_csv(\"data.csv\")\n",
    "print(df1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "df2 = df1.set_index(\"Study ID\", drop = False)\n",
    "#print(df2[\"Study Abstract\"])\n",
    "df3 = df2[\"Study ID\"]\n",
    "#print(df3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Study id: ERP104068\n",
      "Study name: EMG produced TPA metagenomics assembly of the Raw reads of the microbiota of premature infant mouth, skin, and gut (human gut metagenome) data set\n",
      "Study abstract: The human gut metagenome Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA327106. This project includes samples from the following biomes : Human gut.\n",
      "Biome: Human root:Host-associated:Human\n",
      "_____________________________________________________________\n",
      "Study id: SRP002480\n",
      "Study name: Gene-Environment Interactions at the Skin Surface\n",
      "Study abstract: 16S rRNA gene sequences amplified from subjects with eczema and age-matched healthy controls.  Microbes living in and on humans are ten times more numerous than human cells. Culture-based methods have been the primary techniques used to study microbes inhabiting humans; however, many species are not successfully grown in culture. The NIH Roadmap for Medical Research Human Microbiome Project (HMP) aims to investigate the microbes in the gastrointestinal tract, oral cavity, skin, vagina, and nares. The goal of the HMP is to comprehensively characterize the human microbiota and analyze its role in human health and disease.   The skin serves not only as a barrier against invading pathogens and moisture loss, but also as a host to microbial communities. The skin of an adult is an approximately 2 square meter surface with a myriad of microenvironments.  Atopic dermatitis (AD), more commonly known as eczema, is a common skin disorder that is exacerbated by Staphylococcus aureus colonization.  This study aims to investigate the skin microbiota of AD patients at specific timepoints (quiescence, disease flares, and post-treatment) to examine how disease state correlates with changes in the skin microflora.     The 16S small subunit ribosomal (rRNA) gene is present in every bacterial cell, serving as a universal marker. The gene is sufficiently conserved to allow accurate alignment but adequately varied to enable phylogenetic analyses. Sequencing via 16S rRNA-based phylotyping has been used to survey the bacterial microbes on the skin.  Initially, we will examine the bacterial diversity associated with AD using a 16S rRNA survey and later broadening to include fungi, viruses, archaea and mites. In addition, bacteria will be cultured on a wide variety of media and isolates representing both abundant and novel species will be selected for whole genome sequencing.  To characterize fungal diversity, we utilize two phylogenetic markers within the rDNA region: 18S rRNA and the Intervening Internal Transcribed Spacer (ITS) region.  Full microbial diversity will be explored with shotgun metagenomic sequencing.\n",
      "Biome: Skin root:Host-associated:Human:Skin\n",
      "_____________________________________________________________\n",
      "Study id: ERP018577\n",
      "Study name: Human skin bacterial and fungal microbiotas analyses\n",
      "Study abstract: Using high-throughput 16S rDNA and ITS1 sequencing, we characterized skin bacterial and fungal microbiotas from healthy and dandruff subjects, comparing scalp and forehead (lesional and non-lesional skin sites).\n",
      "Biome: Skin root:Host-associated:Human:Skin\n",
      "_____________________________________________________________\n",
      "Study id: ERP022958\n",
      "Study name: Impact of the Mk VI SkinSuit on skin microbiota of terrestrial volunteers and an International Space Station-bound astronaut\n",
      "Study abstract: Microgravity induces physiological deconditioning due to the absence of gravity loading, resulting in bone mineral density loss, atrophy of lower limb skeletal and postural muscles, and lengthening of the spine. SkinSuit is a lightweight compression suit designed to provide head-to-foot (axial) loading to counteract spinal elongation during spaceflight. As synthetic garments may impact negatively on the skin microbiome, we used 16S ribosomal RNA (rRNA) gene amplicon procedures to define bacterial skin communities at sebaceous and moist body sites of five healthy male volunteers undergoing SkinSuit evaluation. Each volunteer displayed a diverse, distinct bacterial population at each skin site. Short (8 h) periods of dry hyper-buoyancy flotation wearing either gym kit or SkinSuit elicited changes in the composition of the skin microbiota at the genus level but had little or no impact on community structure at the phylum level or the richness and diversity of the bacterial population. We also determined the composition of the skin microbiota of an astronaut during pre-flight training, during an eight-day visit to the International Space Station involving two 8 h periods of SkinSuit wear, and for one month after return. Changes in composition of bacterial skin communities at five body sites were strongly linked to changes in geographical location. A distinct ISS microbiome signature was found which reversed to a pre-flight profile on return. No changes in microbiome complexity or diversity were noted, with little evidence for colonisation by potentially pathogenic bacteria. We conclude that short periods of SkinSuit wear do not compromise the healthy skin microbiome.\n",
      "Biome: Skin root:Host-associated:Human:Skin\n",
      "_____________________________________________________________\n",
      "Study id: ERP019566\n",
      "Study name: Longitudinal study of the diabetic skin and wound microbiome\n",
      "Study abstract: Background: Type II diabetes is a chronic health condition which is associated with skin conditions including chronic foot ulcers and an increased incidence of skin infections.  The skin microbiome is thought to play important roles in skin defence and immune functioning.  Diabetes affects the skin environment, and this may perturb skin microbiome with possible implications for skin infections and wound healing.  This study examines the skin and wound microbiome in type II diabetes. Results: Ten type II diabetic subjects with chronic foot ulcers were followed over a time course of 10 weeks, sampling from both foot skin (swabs) and wounds (swabs and debrided tissue) every two weeks. A control group of 8 non-diabetic subjects was also followed over 10 weeks, and skin swabs collected from the foot skin every two weeks.  The diabetic skin microbiome was significantly less diverse than non-diabetic skin.  Community composition was also significantly different between diabetic and non-diabetic skin, however the most abundant taxa were similar between groups, with differences driven by very low abundant members of the skin communities. Chronic wounds were generally colonized by the most dominant skin Staphylococcus, along with other taxa that generally different by patient. No significant correlations were found between wound duration or healing status and the abundance of any particular taxa, while wound size was positively correlated with the dominant skin Staphylococcus.   Conclusions: The major difference observed in this study of the skin microbiome associated with diabetes was a significant reduction in diversity.  The long-term effects of reduced diversity are not yet well understood, but are often associated with disease conditions.\n",
      "Biome: Skin root:Host-associated:Human:Skin\n",
      "_____________________________________________________________\n",
      "Study id: ERP016629\n",
      "Study name: Microbiome samples derived from Buruli ulcer wounds and non-Buruli ulcer skin ulcerations\n",
      "Study abstract: Background: Buruli ulcer (BU) is an infectious disease caused by Mycobacterium ulcerans and considered the third most prevalent mycobacterial disease in humans. This skin disease can affect the entire body surface, but primarily the lower extremities, with symptoms varying from nodules to ulcers. Secondary bacterial infections in open BU lesions are the main cause of pain, delayed healing and systemic illness, resulting in prolonged hospital stay. Thus understanding the diversity of bacteria in these open lesions is important for proper treatment. The normal skin flora is known to protect the host through commensal relationships, however, certain factors such as ulceration can shift the skin flora from being primarily commensal to potentially pathogenic. Therefore, the objective of this study was to characterize and compare the skin bacteria from BU ulcers, non-BU ulcers, and similar locations on the skin of healthy individuals. Results: Using 16S rRNA sequencing, we determined the microbial composition of 5 BU lesions, 3 non-BU lesions and 3 healthy skin samples, and compared these using a data analysis pipeline. Our results showed a lower bacterial diversity in both the BU and non-BU lesions compared to the healthy skin. However, no significant differences were found between BU and non-BU lesions. The BU lesions were characterized by an increase of Bacteroidetes compared to the non-BU wounds, which contained more (Gamma)Proteobacteria. Furthermore, the BU lesions also contained significantly more obligate anaerobes. All lesions contained a mixture of gram-positive and gram-negative bacteria, with more gram-negatives present in the BU and non-BU lesions than in healthy skin samples. With this molecular-based study, we were also able to detect bacteria, which were missed by culture-based methods in previous BU studies.Conclusions: Our study suggests that BU leads to changes in the bacterial community within the lesions. These changes are potentially detrimental and may cause a delay in the healing process. In order to determine if the alterations in Bacteroidetes were due to a specific wound environment, underlying pathophysiological conditions created by M. ulcerans, or associated with wound location on the body, further microbiome studies are necessary.\n",
      "Biome: Skin root:Host-associated:Human:Skin\n",
      "_____________________________________________________________\n",
      "Study id: ERP021525\n",
      "Study name: Micromes on salmon skin and surrounding sea water\n",
      "Study abstract: Pilot study to investigate if sampled Atlantic salmon skin mucus surrounding sea water could prodce quality metagenome sequence reads.\n",
      "Biome: Skin root:Host-associated:Fish:Skin\n",
      "_____________________________________________________________\n",
      "Study id: SRP056364\n",
      "Study name: Skin microbiome in human volunteers inoculated with H. ducreyi Raw sequence reads\n",
      "Study abstract: The aim of this project was to investigate the interaction of the skin microbiome and Haemophilus ducreyi during experimental infection of human volunteers. We hypothesized that the skin microbiome influences susceptibility to H. ducreyi infection and that infection with H. ducreyi causes changes in the skin microbiome. To address these hypotheses, we collected skin swab samples prior to inoculation with H. ducreyi, at Days 1, 2 and 5 post-inoculation, at clinical endpoint, and at the test of cure visit. Samples from four dose-matched pairs of volunteers that either resolved the infection or formed pustules were subjected to barcoded amplification and pyrosequencing of the V1-V3 region of bacterial 16S rRNA genes. We compared alpha and beta diversity measurements of the microbiomes of resolvers and pustule formers, and determined signature bacteria that were associated with infection resolution or pustule formation.\n",
      "Biome: Skin root:Host-associated:Human:Skin\n",
      "_____________________________________________________________\n",
      "Study id: ERP104518\n",
      "Study name: skin microbiota in infected frogs\n",
      "Study abstract: skin microbiota in infected frogs\n",
      "Biome: Amphibia root:Host-associated:Amphibia\n",
      "_____________________________________________________________\n",
      "Study id: ERP104520\n",
      "Study name: Skin microbiota of Scinax alcatraz\n",
      "Study abstract: Skin microbiota of Scinax alcatraz\n",
      "Biome: Amphibia root:Host-associated:Amphibia\n",
      "_____________________________________________________________\n",
      "Study id: ERP104516\n",
      "Study name: Variations on the diversity of amphibian skin microbiota across forests fragments\n",
      "Study abstract: Variations on the diversity of amphibian skin microbiota across forests fragments\n",
      "Biome: Amphibia root:Host-associated:Amphibia\n",
      "_____________________________________________________________\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(df3)):\n",
    "    with Session(API_BASE) as s:\n",
    "        study = s.get('studies', df3[i]).resource\n",
    "        print('Study id:', study.id)\n",
    "        print('Study name:', study.study_name)\n",
    "        print('Study abstract:', study.study_abstract)\n",
    "        for biome in study.biomes:\n",
    "            print('Biome:', biome.biome_name, biome.lineage)\n",
    "        print('_____________________________________________________________')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "study1 = 'SRP002480'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### List samples with biomes for the given study\n",
    "\n",
    "Get study: https://www.ebi.ac.uk/metagenomics/api/latest/studies/SRP002480\n",
    "\n",
    "List samples: https://www.ebi.ac.uk/metagenomics/api/latest/studies/SRP002480/samples\n",
    "\n",
    "\n",
    "Fetch samples for the given study accession: https://www.ebi.ac.uk/metagenomics/api/latest/samples?study_accession=SRP002480\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "data_sample_from_studySRP002480.csv\n"
     ]
    }
   ],
   "source": [
    "filename = 'data_sample_from_study'+study1+'.csv'\n",
    "print(filename)\n",
    "if not os.path.isfile(filename):\n",
    "    with open(filename, 'wb') as f:\n",
    "        c = pycurl.Curl()\n",
    "        c.setopt(c.URL, 'https://www.ebi.ac.uk/metagenomics/projects/'+study1+'/overview/doExport')\n",
    "        c.setopt(c.WRITEDATA, f)\n",
    "        c.perform()\n",
    "        c.close()\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Release version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRS451417</td>\n",
       "      <td>SRR919527</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRS451417</td>\n",
       "      <td>SRR919587</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRS451418</td>\n",
       "      <td>SRR919528</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRS451418</td>\n",
       "      <td>SRR919588</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRS451419</td>\n",
       "      <td>SRR919529</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SRS451419</td>\n",
       "      <td>SRR919589</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SRS451420</td>\n",
       "      <td>SRR919530</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRS451420</td>\n",
       "      <td>SRR919590</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SRS451421</td>\n",
       "      <td>SRR919531</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SRS451421</td>\n",
       "      <td>SRR919591</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SRS451422</td>\n",
       "      <td>SRR919532</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SRS451422</td>\n",
       "      <td>SRR919592</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SRS451423</td>\n",
       "      <td>SRR919533</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SRS451423</td>\n",
       "      <td>SRR919593</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SRS451424</td>\n",
       "      <td>SRR919534</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SRS451424</td>\n",
       "      <td>SRR919594</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SRS451425</td>\n",
       "      <td>SRR919535</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SRS451425</td>\n",
       "      <td>SRR919595</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SRS451426</td>\n",
       "      <td>SRR919537</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SRS451426</td>\n",
       "      <td>SRR919597</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SRS451427</td>\n",
       "      <td>SRR919536</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SRS451427</td>\n",
       "      <td>SRR919596</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SRS451428</td>\n",
       "      <td>SRR919538</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SRS451428</td>\n",
       "      <td>SRR919598</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SRS451429</td>\n",
       "      <td>SRR919539</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SRS451429</td>\n",
       "      <td>SRR919599</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SRS451430</td>\n",
       "      <td>SRR919540</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SRS451430</td>\n",
       "      <td>SRR919600</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SRS451431</td>\n",
       "      <td>SRR919541</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SRS451431</td>\n",
       "      <td>SRR919601</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4368</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633154</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4369</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633155</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4370</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633156</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4371</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633157</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4372</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633158</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4373</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633147</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4374</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633148</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4375</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633149</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4376</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633150</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4377</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633151</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4378</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633152</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4379</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633159</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4380</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633160</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4381</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633161</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4382</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633162</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4383</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633163</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4384</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633164</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4385</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633165</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4386</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633166</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4387</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633167</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4388</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633168</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4389</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633169</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4390</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633170</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4391</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633171</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4392</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633172</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4393</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633173</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4394</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633174</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4395</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633175</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4396</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633176</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4397</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633177</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4398 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Sample ID      Run ID  Release version\n",
       "0     SRS451417   SRR919527              2.0\n",
       "1     SRS451417   SRR919587              2.0\n",
       "2     SRS451418   SRR919528              2.0\n",
       "3     SRS451418   SRR919588              2.0\n",
       "4     SRS451419   SRR919529              2.0\n",
       "5     SRS451419   SRR919589              2.0\n",
       "6     SRS451420   SRR919530              2.0\n",
       "7     SRS451420   SRR919590              2.0\n",
       "8     SRS451421   SRR919531              2.0\n",
       "9     SRS451421   SRR919591              2.0\n",
       "10    SRS451422   SRR919532              2.0\n",
       "11    SRS451422   SRR919592              2.0\n",
       "12    SRS451423   SRR919533              2.0\n",
       "13    SRS451423   SRR919593              2.0\n",
       "14    SRS451424   SRR919534              2.0\n",
       "15    SRS451424   SRR919594              2.0\n",
       "16    SRS451425   SRR919535              2.0\n",
       "17    SRS451425   SRR919595              2.0\n",
       "18    SRS451426   SRR919537              2.0\n",
       "19    SRS451426   SRR919597              2.0\n",
       "20    SRS451427   SRR919536              2.0\n",
       "21    SRS451427   SRR919596              2.0\n",
       "22    SRS451428   SRR919538              2.0\n",
       "23    SRS451428   SRR919598              2.0\n",
       "24    SRS451429   SRR919539              2.0\n",
       "25    SRS451429   SRR919599              2.0\n",
       "26    SRS451430   SRR919540              2.0\n",
       "27    SRS451430   SRR919600              2.0\n",
       "28    SRS451431   SRR919541              2.0\n",
       "29    SRS451431   SRR919601              2.0\n",
       "...         ...         ...              ...\n",
       "4368  SRS732139  SRR1633154              2.0\n",
       "4369  SRS732139  SRR1633155              2.0\n",
       "4370  SRS732139  SRR1633156              2.0\n",
       "4371  SRS732139  SRR1633157              2.0\n",
       "4372  SRS732139  SRR1633158              2.0\n",
       "4373  SRS732140  SRR1633147              2.0\n",
       "4374  SRS732140  SRR1633148              2.0\n",
       "4375  SRS732140  SRR1633149              2.0\n",
       "4376  SRS732140  SRR1633150              2.0\n",
       "4377  SRS732140  SRR1633151              2.0\n",
       "4378  SRS732140  SRR1633152              2.0\n",
       "4379  SRS732141  SRR1633159              2.0\n",
       "4380  SRS732141  SRR1633160              2.0\n",
       "4381  SRS732141  SRR1633161              2.0\n",
       "4382  SRS732141  SRR1633162              2.0\n",
       "4383  SRS732141  SRR1633163              2.0\n",
       "4384  SRS732141  SRR1633164              2.0\n",
       "4385  SRS732141  SRR1633165              2.0\n",
       "4386  SRS732142  SRR1633166              2.0\n",
       "4387  SRS732142  SRR1633167              2.0\n",
       "4388  SRS732142  SRR1633168              2.0\n",
       "4389  SRS732142  SRR1633169              2.0\n",
       "4390  SRS732142  SRR1633170              2.0\n",
       "4391  SRS732142  SRR1633171              2.0\n",
       "4392  SRS732144  SRR1633172              2.0\n",
       "4393  SRS732144  SRR1633173              2.0\n",
       "4394  SRS732144  SRR1633174              2.0\n",
       "4395  SRS732144  SRR1633175              2.0\n",
       "4396  SRS732144  SRR1633176              2.0\n",
       "4397  SRS732144  SRR1633177              2.0\n",
       "\n",
       "[4398 rows x 3 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Script ini untuk mengekstrak data tabel dari file CSV\n",
    "df_sample_list = pd.read_csv(\"data_sample_from_studySRP002480.csv\")\n",
    "df_sample_list_refine = df_sample_list[[\"Sample ID\",\"Run ID\",\"Release version\"]]\n",
    "df_sample_list_refine\n",
    "#df_sample_list "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'SRS451417'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sample_list_refine.loc[0, \"Sample ID\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sex</th>\n",
       "      <th>Body site</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>male</td>\n",
       "      <td>external auditory canal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>male</td>\n",
       "      <td>external auditory canal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>male</td>\n",
       "      <td>glabella</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>male</td>\n",
       "      <td>glabella</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>male</td>\n",
       "      <td>inguinal crease</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Sex                 Body site\n",
       "No                                 \n",
       "0   male        antecubital crease \n",
       "1   male        antecubital crease \n",
       "2   male                      back \n",
       "3   male                      back \n",
       "4   male   external auditory canal \n",
       "5   male   external auditory canal \n",
       "6   male           hypothenar palm \n",
       "7   male           hypothenar palm \n",
       "8   male     retroauricular crease \n",
       "9   male     retroauricular crease \n",
       "10  male                    toeweb \n",
       "11  male                    toeweb \n",
       "12  male     retroauricular crease \n",
       "13  male     retroauricular crease \n",
       "14  male                    toeweb \n",
       "15  male                    toeweb \n",
       "16  male        antecubital crease \n",
       "17  male        antecubital crease \n",
       "18  male                  glabella \n",
       "19  male                  glabella \n",
       "20  male                      back \n",
       "21  male                      back \n",
       "22  male           hypothenar palm \n",
       "23  male           hypothenar palm \n",
       "24  male           inguinal crease "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_metadata(metadata, key):\n",
    "    import html\n",
    "    for m in metadata:\n",
    "        if m['key'].lower() == key.lower():\n",
    "            value = m['value']\n",
    "            unit = html.unescape(m['unit']) if m['unit'] else \"\"\n",
    "            return \"{value} {unit}\".format(value=value, unit=unit)\n",
    "    return None\n",
    "\n",
    "df = DataFrame(columns=('Sex',\"Body site\"))\n",
    "df.index.name = 'No'\n",
    "\n",
    "#for i in range(len(df_sample_list_refine)):\n",
    "for i in range(25):\n",
    "    with Session(API_BASE) as s:\n",
    "        study = s.get('samples', df_sample_list_refine.loc[i, \"Sample ID\"]).resource\n",
    "        #print(get_metadata(study.sample_metadata, 'sex'))\n",
    "        df.loc[i] = [\n",
    "            get_metadata(study.sample_metadata, 'sex'),\n",
    "            get_metadata(study.sample_metadata, 'body site')\n",
    "        ]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Release version</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Body site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRS451417</td>\n",
       "      <td>SRR919527</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRS451417</td>\n",
       "      <td>SRR919587</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRS451418</td>\n",
       "      <td>SRR919528</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRS451418</td>\n",
       "      <td>SRR919588</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRS451419</td>\n",
       "      <td>SRR919529</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>external auditory canal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SRS451419</td>\n",
       "      <td>SRR919589</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>external auditory canal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SRS451420</td>\n",
       "      <td>SRR919530</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRS451420</td>\n",
       "      <td>SRR919590</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SRS451421</td>\n",
       "      <td>SRR919531</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SRS451421</td>\n",
       "      <td>SRR919591</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SRS451422</td>\n",
       "      <td>SRR919532</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SRS451422</td>\n",
       "      <td>SRR919592</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SRS451423</td>\n",
       "      <td>SRR919533</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SRS451423</td>\n",
       "      <td>SRR919593</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SRS451424</td>\n",
       "      <td>SRR919534</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SRS451424</td>\n",
       "      <td>SRR919594</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SRS451425</td>\n",
       "      <td>SRR919535</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SRS451425</td>\n",
       "      <td>SRR919595</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SRS451426</td>\n",
       "      <td>SRR919537</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>glabella</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SRS451426</td>\n",
       "      <td>SRR919597</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>glabella</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SRS451427</td>\n",
       "      <td>SRR919536</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SRS451427</td>\n",
       "      <td>SRR919596</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SRS451428</td>\n",
       "      <td>SRR919538</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SRS451428</td>\n",
       "      <td>SRR919598</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SRS451429</td>\n",
       "      <td>SRR919539</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>inguinal crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SRS451429</td>\n",
       "      <td>SRR919599</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SRS451430</td>\n",
       "      <td>SRR919540</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SRS451430</td>\n",
       "      <td>SRR919600</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SRS451431</td>\n",
       "      <td>SRR919541</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SRS451431</td>\n",
       "      <td>SRR919601</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4368</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633154</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4369</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633155</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4370</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633156</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4371</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633157</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4372</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633158</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4373</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633147</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4374</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633148</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4375</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633149</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4376</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633150</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4377</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633151</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4378</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633152</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4379</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633159</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4380</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633160</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4381</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633161</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4382</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633162</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4383</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633163</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4384</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633164</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4385</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633165</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4386</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633166</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4387</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633167</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4388</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633168</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4389</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633169</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4390</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633170</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4391</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633171</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4392</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633172</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4393</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633173</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4394</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633174</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4395</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633175</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4396</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633176</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4397</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633177</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4398 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Sample ID      Run ID  Release version    Sex                 Body site\n",
       "0     SRS451417   SRR919527              2.0  male        antecubital crease \n",
       "1     SRS451417   SRR919587              2.0  male        antecubital crease \n",
       "2     SRS451418   SRR919528              2.0  male                      back \n",
       "3     SRS451418   SRR919588              2.0  male                      back \n",
       "4     SRS451419   SRR919529              2.0  male   external auditory canal \n",
       "5     SRS451419   SRR919589              2.0  male   external auditory canal \n",
       "6     SRS451420   SRR919530              2.0  male           hypothenar palm \n",
       "7     SRS451420   SRR919590              2.0  male           hypothenar palm \n",
       "8     SRS451421   SRR919531              2.0  male     retroauricular crease \n",
       "9     SRS451421   SRR919591              2.0  male     retroauricular crease \n",
       "10    SRS451422   SRR919532              2.0  male                    toeweb \n",
       "11    SRS451422   SRR919592              2.0  male                    toeweb \n",
       "12    SRS451423   SRR919533              2.0  male     retroauricular crease \n",
       "13    SRS451423   SRR919593              2.0  male     retroauricular crease \n",
       "14    SRS451424   SRR919534              2.0  male                    toeweb \n",
       "15    SRS451424   SRR919594              2.0  male                    toeweb \n",
       "16    SRS451425   SRR919535              2.0  male        antecubital crease \n",
       "17    SRS451425   SRR919595              2.0  male        antecubital crease \n",
       "18    SRS451426   SRR919537              2.0  male                  glabella \n",
       "19    SRS451426   SRR919597              2.0  male                  glabella \n",
       "20    SRS451427   SRR919536              2.0  male                      back \n",
       "21    SRS451427   SRR919596              2.0  male                      back \n",
       "22    SRS451428   SRR919538              2.0  male           hypothenar palm \n",
       "23    SRS451428   SRR919598              2.0  male           hypothenar palm \n",
       "24    SRS451429   SRR919539              2.0  male           inguinal crease \n",
       "25    SRS451429   SRR919599              2.0    NaN                       NaN\n",
       "26    SRS451430   SRR919540              2.0    NaN                       NaN\n",
       "27    SRS451430   SRR919600              2.0    NaN                       NaN\n",
       "28    SRS451431   SRR919541              2.0    NaN                       NaN\n",
       "29    SRS451431   SRR919601              2.0    NaN                       NaN\n",
       "...         ...         ...              ...    ...                       ...\n",
       "4368  SRS732139  SRR1633154              2.0    NaN                       NaN\n",
       "4369  SRS732139  SRR1633155              2.0    NaN                       NaN\n",
       "4370  SRS732139  SRR1633156              2.0    NaN                       NaN\n",
       "4371  SRS732139  SRR1633157              2.0    NaN                       NaN\n",
       "4372  SRS732139  SRR1633158              2.0    NaN                       NaN\n",
       "4373  SRS732140  SRR1633147              2.0    NaN                       NaN\n",
       "4374  SRS732140  SRR1633148              2.0    NaN                       NaN\n",
       "4375  SRS732140  SRR1633149              2.0    NaN                       NaN\n",
       "4376  SRS732140  SRR1633150              2.0    NaN                       NaN\n",
       "4377  SRS732140  SRR1633151              2.0    NaN                       NaN\n",
       "4378  SRS732140  SRR1633152              2.0    NaN                       NaN\n",
       "4379  SRS732141  SRR1633159              2.0    NaN                       NaN\n",
       "4380  SRS732141  SRR1633160              2.0    NaN                       NaN\n",
       "4381  SRS732141  SRR1633161              2.0    NaN                       NaN\n",
       "4382  SRS732141  SRR1633162              2.0    NaN                       NaN\n",
       "4383  SRS732141  SRR1633163              2.0    NaN                       NaN\n",
       "4384  SRS732141  SRR1633164              2.0    NaN                       NaN\n",
       "4385  SRS732141  SRR1633165              2.0    NaN                       NaN\n",
       "4386  SRS732142  SRR1633166              2.0    NaN                       NaN\n",
       "4387  SRS732142  SRR1633167              2.0    NaN                       NaN\n",
       "4388  SRS732142  SRR1633168              2.0    NaN                       NaN\n",
       "4389  SRS732142  SRR1633169              2.0    NaN                       NaN\n",
       "4390  SRS732142  SRR1633170              2.0    NaN                       NaN\n",
       "4391  SRS732142  SRR1633171              2.0    NaN                       NaN\n",
       "4392  SRS732144  SRR1633172              2.0    NaN                       NaN\n",
       "4393  SRS732144  SRR1633173              2.0    NaN                       NaN\n",
       "4394  SRS732144  SRR1633174              2.0    NaN                       NaN\n",
       "4395  SRS732144  SRR1633175              2.0    NaN                       NaN\n",
       "4396  SRS732144  SRR1633176              2.0    NaN                       NaN\n",
       "4397  SRS732144  SRR1633177              2.0    NaN                       NaN\n",
       "\n",
       "[4398 rows x 5 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result = pd.concat([df_sample_list_refine, df], axis=1)\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('List Sample'+study1+'.csv',index=True,header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "ename": "DocumentError",
     "evalue": "Error 404: ?",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mDocumentError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-27-9af9f53d5e57>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     20\u001b[0m     }\n\u001b[1;32m     21\u001b[0m     \u001b[0mf\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mFilter\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0murlencode\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mparams\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m---> 22\u001b[0;31m     \u001b[1;32mfor\u001b[0m \u001b[0msample\u001b[0m \u001b[1;32min\u001b[0m \u001b[0ms\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0miterate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'study'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     23\u001b[0m         df.loc[sample.accession] = [\n\u001b[1;32m     24\u001b[0m             \u001b[0msample\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msample_name\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msample\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbiome\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mid\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\anaconda\\lib\\site-packages\\jsonapi_client\\session.py\u001b[0m in \u001b[0;36m_iterate_sync\u001b[0;34m(self, resource_type, filter)\u001b[0m\n\u001b[1;32m    364\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_iterate_sync\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mresource_type\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfilter\u001b[0m\u001b[1;33m:\u001b[0m \u001b[1;34m'Filter'\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[0;31m \u001b[0m\u001b[0;31m\\\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    365\u001b[0m             \u001b[1;33m->\u001b[0m \u001b[1;34m'Iterator[ResourceObject]'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 366\u001b[0;31m         \u001b[0mdoc\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresource_type\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfilter\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    367\u001b[0m         \u001b[1;32myield\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0mdoc\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_iterator_sync\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    368\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\anaconda\\lib\\site-packages\\jsonapi_client\\session.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, resource_type, resource_id_or_filter)\u001b[0m\n\u001b[1;32m    360\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_async\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresource_type\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mresource_id_or_filter\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    361\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 362\u001b[0;31m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_sync\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresource_type\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mresource_id_or_filter\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    363\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    364\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_iterate_sync\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mresource_type\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfilter\u001b[0m\u001b[1;33m:\u001b[0m \u001b[1;34m'Filter'\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[0;31m \u001b[0m\u001b[0;31m\\\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\anaconda\\lib\\site-packages\\jsonapi_client\\session.py\u001b[0m in \u001b[0;36m_get_sync\u001b[0;34m(self, resource_type, resource_id_or_filter)\u001b[0m\n\u001b[1;32m    336\u001b[0m                                                                 resource_id_or_filter)\n\u001b[1;32m    337\u001b[0m         \u001b[0murl\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_url_for_resource\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresource_type\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mresource_id\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfilter_\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 338\u001b[0;31m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfetch_document_by_url\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    339\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    340\u001b[0m     async def _get_async(self, resource_type: str,\n",
      "\u001b[0;32mC:\\anaconda\\lib\\site-packages\\jsonapi_client\\session.py\u001b[0m in \u001b[0;36mfetch_document_by_url\u001b[0;34m(self, url)\u001b[0m\n\u001b[1;32m    453\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    454\u001b[0m         \u001b[1;31m# TODO: should we try to guess type, id from url?\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 455\u001b[0;31m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdocuments_by_link\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_ext_fetch_by_url\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    456\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    457\u001b[0m     \u001b[1;33masync\u001b[0m \u001b[1;32mdef\u001b[0m \u001b[0mfetch_document_by_url_async\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[1;34m'Document'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\anaconda\\lib\\site-packages\\jsonapi_client\\session.py\u001b[0m in \u001b[0;36m_ext_fetch_by_url\u001b[0;34m(self, url)\u001b[0m\n\u001b[1;32m    467\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    468\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_ext_fetch_by_url\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[1;34m'Document'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 469\u001b[0;31m         \u001b[0mjson_data\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_fetch_json\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    470\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mjson_data\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    471\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\anaconda\\lib\\site-packages\\jsonapi_client\\session.py\u001b[0m in \u001b[0;36m_fetch_json\u001b[0;34m(self, url)\u001b[0m\n\u001b[1;32m    492\u001b[0m                                 \u001b[1;34mf'{error_from_response(response)}'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    493\u001b[0m                                 \u001b[0merrors\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'status_code'\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mresponse\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstatus_code\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 494\u001b[0;31m                                 response=response)\n\u001b[0m\u001b[1;32m    495\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    496\u001b[0m     \u001b[1;33masync\u001b[0m \u001b[1;32mdef\u001b[0m \u001b[0m_fetch_json_async\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[0mdict\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mDocumentError\u001b[0m: Error 404: ?"
     ]
    }
   ],
   "source": [
    "def get_metadata(metadata, key):\n",
    "    import html\n",
    "    for m in metadata:\n",
    "        if m['key'].lower() == key.lower():\n",
    "            value = m['value']\n",
    "            unit = html.unescape(m['unit']) if m['unit'] else \"\"\n",
    "            return \"{value} {unit}\".format(value=value, unit=unit)\n",
    "    return None\n",
    "\n",
    "depth_label = 'geographic location (depth)'\n",
    "temp_label = 'temperature'\n",
    "df = DataFrame(columns=('sample name', 'biome', 'temperature', 'depth', 'longitude', 'latitude'))\n",
    "df.index.name = 'accession'\n",
    "\n",
    "with Session(API_BASE) as s:\n",
    "    params = {\n",
    "        'study_accession': study1,\n",
    "        'include': 'biome',\n",
    "        'page_size': 100,\n",
    "    }\n",
    "    f = Filter(urlencode(params))\n",
    "    for sample in s.iterate('study', f):\n",
    "        df.loc[sample.accession] = [\n",
    "            sample.sample_name, sample.biome.id,\n",
    "            get_metadata(sample.sample_metadata, temp_label),\n",
    "            get_metadata(sample.sample_metadata, depth_label),\n",
    "            sample.longitude, sample.latitude\n",
    "        ]\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sampling Data BIOM dari EBI\n",
    "https://www.ebi.ac.uk/metagenomics//projects/SRP002480/samples/SRS451457/runs/SRR919567/results/versions/2.0/taxonomy/OTU-table-HDF5-BIOM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Sample ID      Run ID\n",
      "0     SRS451417   SRR919527\n",
      "1     SRS451417   SRR919587\n",
      "2     SRS451418   SRR919528\n",
      "3     SRS451418   SRR919588\n",
      "4     SRS451419   SRR919529\n",
      "5     SRS451419   SRR919589\n",
      "6     SRS451420   SRR919530\n",
      "7     SRS451420   SRR919590\n",
      "8     SRS451421   SRR919531\n",
      "9     SRS451421   SRR919591\n",
      "10    SRS451422   SRR919532\n",
      "11    SRS451422   SRR919592\n",
      "12    SRS451423   SRR919533\n",
      "13    SRS451423   SRR919593\n",
      "14    SRS451424   SRR919534\n",
      "15    SRS451424   SRR919594\n",
      "16    SRS451425   SRR919535\n",
      "17    SRS451425   SRR919595\n",
      "18    SRS451426   SRR919537\n",
      "19    SRS451426   SRR919597\n",
      "20    SRS451427   SRR919536\n",
      "21    SRS451427   SRR919596\n",
      "22    SRS451428   SRR919538\n",
      "23    SRS451428   SRR919598\n",
      "24    SRS451429   SRR919539\n",
      "25    SRS451429   SRR919599\n",
      "26    SRS451430   SRR919540\n",
      "27    SRS451430   SRR919600\n",
      "28    SRS451431   SRR919541\n",
      "29    SRS451431   SRR919601\n",
      "...         ...         ...\n",
      "4368  SRS732139  SRR1633154\n",
      "4369  SRS732139  SRR1633155\n",
      "4370  SRS732139  SRR1633156\n",
      "4371  SRS732139  SRR1633157\n",
      "4372  SRS732139  SRR1633158\n",
      "4373  SRS732140  SRR1633147\n",
      "4374  SRS732140  SRR1633148\n",
      "4375  SRS732140  SRR1633149\n",
      "4376  SRS732140  SRR1633150\n",
      "4377  SRS732140  SRR1633151\n",
      "4378  SRS732140  SRR1633152\n",
      "4379  SRS732141  SRR1633159\n",
      "4380  SRS732141  SRR1633160\n",
      "4381  SRS732141  SRR1633161\n",
      "4382  SRS732141  SRR1633162\n",
      "4383  SRS732141  SRR1633163\n",
      "4384  SRS732141  SRR1633164\n",
      "4385  SRS732141  SRR1633165\n",
      "4386  SRS732142  SRR1633166\n",
      "4387  SRS732142  SRR1633167\n",
      "4388  SRS732142  SRR1633168\n",
      "4389  SRS732142  SRR1633169\n",
      "4390  SRS732142  SRR1633170\n",
      "4391  SRS732142  SRR1633171\n",
      "4392  SRS732144  SRR1633172\n",
      "4393  SRS732144  SRR1633173\n",
      "4394  SRS732144  SRR1633174\n",
      "4395  SRS732144  SRR1633175\n",
      "4396  SRS732144  SRR1633176\n",
      "4397  SRS732144  SRR1633177\n",
      "\n",
      "[4398 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df4 = pd.read_csv(\"skin.csv\")\n",
    "df5 = df4.loc[:,\"Sample ID\":\"Run ID\"]\n",
    "print(df5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0     SRS451417\n",
      "1     SRS451417\n",
      "2     SRS451418\n",
      "3     SRS451418\n",
      "4     SRS451419\n",
      "5     SRS451419\n",
      "6     SRS451420\n",
      "7     SRS451420\n",
      "8     SRS451421\n",
      "9     SRS451421\n",
      "10    SRS451422\n",
      "11    SRS451422\n",
      "12    SRS451423\n",
      "13    SRS451423\n",
      "14    SRS451424\n",
      "15    SRS451424\n",
      "16    SRS451425\n",
      "17    SRS451425\n",
      "18    SRS451426\n",
      "19    SRS451426\n",
      "20    SRS451427\n",
      "21    SRS451427\n",
      "22    SRS451428\n",
      "23    SRS451428\n",
      "24    SRS451429\n",
      "25    SRS451429\n",
      "Name: Sample ID, dtype: object\n"
     ]
    }
   ],
   "source": [
    "data_sampel = df5.loc[0:25,'Sample ID']\n",
    "print(data_sampel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0     SRR919527\n",
      "1     SRR919587\n",
      "2     SRR919528\n",
      "3     SRR919588\n",
      "4     SRR919529\n",
      "5     SRR919589\n",
      "6     SRR919530\n",
      "7     SRR919590\n",
      "8     SRR919531\n",
      "9     SRR919591\n",
      "10    SRR919532\n",
      "11    SRR919592\n",
      "12    SRR919533\n",
      "13    SRR919593\n",
      "14    SRR919534\n",
      "15    SRR919594\n",
      "16    SRR919535\n",
      "17    SRR919595\n",
      "18    SRR919537\n",
      "19    SRR919597\n",
      "20    SRR919536\n",
      "21    SRR919596\n",
      "22    SRR919538\n",
      "23    SRR919598\n",
      "24    SRR919539\n",
      "25    SRR919599\n",
      "Name: Run ID, dtype: object\n"
     ]
    }
   ],
   "source": [
    "data_run = df5.loc[0:25,'Run ID']\n",
    "print(data_run)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "D:\\Jupyter\\KetiakProject\\src\n"
     ]
    }
   ],
   "source": [
    "#Create ouput folder\n",
    "import os, sys\n",
    "\n",
    "cwd = os.getcwd()\n",
    "output_folder = \"\\output\" #name output folder, use \\ for directory\n",
    "\n",
    "if not os.path.isdir(cwd + output_folder):\n",
    "    os.mkdir(cwd + output_folder)\n",
    "\n",
    "new_dir = cwd + output_folder\n",
    "    \n",
    "print(cwd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRS451417_SRR919527.biom\n",
      "SRS451417_SRR919587.biom\n",
      "SRS451418_SRR919528.biom\n",
      "SRS451418_SRR919588.biom\n",
      "SRS451419_SRR919529.biom\n",
      "SRS451419_SRR919589.biom\n",
      "SRS451420_SRR919530.biom\n",
      "SRS451420_SRR919590.biom\n",
      "SRS451421_SRR919531.biom\n",
      "SRS451421_SRR919591.biom\n",
      "SRS451422_SRR919532.biom\n",
      "SRS451422_SRR919592.biom\n",
      "SRS451423_SRR919533.biom\n",
      "SRS451423_SRR919593.biom\n",
      "SRS451424_SRR919534.biom\n",
      "SRS451424_SRR919594.biom\n",
      "SRS451425_SRR919535.biom\n",
      "SRS451425_SRR919595.biom\n",
      "SRS451426_SRR919537.biom\n",
      "SRS451426_SRR919597.biom\n",
      "SRS451427_SRR919536.biom\n",
      "SRS451427_SRR919596.biom\n",
      "SRS451428_SRR919538.biom\n",
      "SRS451428_SRR919598.biom\n",
      "SRS451429_SRR919539.biom\n",
      "done\n"
     ]
    }
   ],
   "source": [
    "import pycurl\n",
    "\n",
    "#Ambil data dari EBI\n",
    "for i in range(25):\n",
    "    os.chdir(new_dir) #pindah ke folder output\n",
    "    filename = data_sampel[i] + '_' + data_run[i] + '.biom'\n",
    "    print(filename)\n",
    "    if not os.path.isfile(filename):\n",
    "        with open(filename, 'wb') as f:\n",
    "            c = pycurl.Curl()\n",
    "            c.setopt(c.URL, 'https://www.ebi.ac.uk/metagenomics//projects/SRP002480/samples/'+ data_sampel[i] + '/runs/' + data_run[i] +'/results/versions/2.0/taxonomy/OTU-table-JSON-BIOM')\n",
    "            c.setopt(c.WRITEDATA, f)\n",
    "            c.perform()\n",
    "            c.close()\n",
    "    os.chdir(cwd) #balik ke folder semula\n",
    "print('done')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### References\n",
    "---\n",
    "<a id='ref1'></a>\n",
    "1. Alex L Mitchell, Maxim Scheremetjew, Hubert Denise, Simon Potter, Aleksandra Tarkowska, Matloob Qureshi, Gustavo A Salazar, Sebastien Pesseat, Miguel A Boland, Fiona M I Hunter, Petra ten Hoopen, Blaise Alako, Clara Amid, Darren J Wilkinson, Thomas P Curtis, Guy Cochrane, Robert D Finn; EBI Metagenomics in 2017: enriching the analysis of microbial communities, from sequence reads to assemblies, Nucleic Acids Research, Volume 46, Issue D1, 4 January 2018, Pages D726–D735, https://doi.org/10.1093/nar/gkx967"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
