{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Mining of Human Skin Microbiome from EBI-Metagenomics Portal\n",
    "\n",
    "_Matin Nuhamunada_<sup>1*</sup>, _Gregorius Altius Pratama_<sup>1</sup>, _Setianing Wikanthi_<sup>2</sup>, and _Mohamad Khoirul Anam_<sup>1</sup>\n",
    "\n",
    "<sup>1</sup>Department of Tropical Biology, Universitas Gadjah Mada;   \n",
    "Jl. Teknika Selatan, Sekip Utara, Bulaksumur, Yogyakarta, Indonesia, 55281;   \n",
    "\n",
    "<sup>2</sup>Department of Agricultural Microbiology, Universitas Gadjah Mada;  \n",
    "\n",
    "*Correspondence: [matin_nuhamunada@ugm.ac.id](mailto:matin_nuhamunada@mail.ugm.ac.id)  \n",
    "[mohamad.khoirul.anam@mail.ugm.ac.id](mailto:mohamad.khoirul.anam@mail.ugm.ac.id)  \n",
    "[gregorius.altius.p@mail.ugm.ac.id](mailto:gregorius.altius.p@mail.ugm.ac.id)  \n",
    "[setianingwikanthi@mail.ugm.ac.id](mailto:setianingwikanthi@mail.ugm.ac.id)\n",
    "\n",
    "---\n",
    "## Abstract\n",
    "Human skin microbiome is unique to individuals in regards to many aspects, including behaviour, environment, and perhaps maybe genes. To understand more about the distribution of human skin microbiome across the globe, we compare several skin microbiome study available in the EBI-Metagenomic Portal. Study data was acquired using EBI-Metagenome API, and sample data was selected based on sex, location, and bodysite. The biological observation matrix from the analysis result of the selected samples were compared using MEGAN. \n",
    "\n",
    "### Keywords\n",
    "Human Skin, Microbiome, EBI-Metagenome\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import Python Modules\n",
    "We use python 3 script using ``pandas``, ``jsonapi_client``, ``pycurl``, to mine the data from EBI-metagenomic portal  [[1]](#ref1)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import DataFrame\n",
    "import pandas as pd\n",
    "\n",
    "try:\n",
    "    from urllib import urlencode\n",
    "except ImportError:\n",
    "    from urllib.parse import urlencode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from jsonapi_client import Session, Filter\n",
    "\n",
    "API_BASE = 'https://www.ebi.ac.uk/metagenomics/api/latest/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pycurl\n",
    "import os, sys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get Study\n",
    "We search the EBI Metagenomic database on human skin microbiome study in the host-associated biome with 'skin' as search term. The study list can be found on this link: https://www.ebi.ac.uk/metagenomics/projects/doExportDetails?searchTerm=skin&includingChildren=true&biomeLineage=root%3AHost-associated%3AHuman&search=Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Study Data.csv\n"
     ]
    }
   ],
   "source": [
    "filename = 'Study Data.csv'\n",
    "print(filename)\n",
    "if not os.path.isfile(filename):\n",
    "    with open(filename, 'wb') as f:\n",
    "        c = pycurl.Curl()\n",
    "        c.setopt(c.URL, 'https://www.ebi.ac.uk/metagenomics/projects/doExportDetails?searchTerm=skin&includingChildren=true&biomeLineage=root%3AHost-associated%3AHuman&search=Search')\n",
    "        c.setopt(c.WRITEDATA, f)\n",
    "        c.perform()\n",
    "        c.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load Study Data\n",
    "df1 = pd.read_csv(\"data.csv\")\n",
    "#df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Select relevant information\n",
    "study_data = df1[[\"Study ID\",\"Study Name\",\"Number Of Samples\", \"Study Abstract\"]]\n",
    "#study_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Add information on biome\n",
    "df2 = DataFrame(columns=(\"Biome\",\"Lineage\"))\n",
    "df2.index.name = 'No'\n",
    "\n",
    "for i in range(len(study_data)):\n",
    "    with Session(API_BASE) as s:\n",
    "        study = s.get('studies', study_data.loc[i, \"Study ID\"]).resource\n",
    "        for biome in study.biomes:\n",
    "            df2.loc[i] = [biome.biome_name,\n",
    "                          biome.lineage\n",
    "        ]\n",
    "#df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study ID</th>\n",
       "      <th>Study Name</th>\n",
       "      <th>Number Of Samples</th>\n",
       "      <th>Study Abstract</th>\n",
       "      <th>Biome</th>\n",
       "      <th>Lineage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERP104068</td>\n",
       "      <td>EMG produced TPA metagenomics assembly of the ...</td>\n",
       "      <td>45</td>\n",
       "      <td>The human gut metagenome Third Party Annotatio...</td>\n",
       "      <td>Human</td>\n",
       "      <td>root:Host-associated:Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRP002480</td>\n",
       "      <td>Gene-Environment Interactions at the Skin Surface</td>\n",
       "      <td>2560</td>\n",
       "      <td>16S rRNA gene sequences amplified from subject...</td>\n",
       "      <td>Skin</td>\n",
       "      <td>root:Host-associated:Human:Skin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERP018577</td>\n",
       "      <td>Human skin bacterial and fungal microbiotas an...</td>\n",
       "      <td>96</td>\n",
       "      <td>Using high-throughput 16S rDNA and ITS1 sequen...</td>\n",
       "      <td>Skin</td>\n",
       "      <td>root:Host-associated:Human:Skin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERP022958</td>\n",
       "      <td>Impact of the Mk VI SkinSuit on skin microbiot...</td>\n",
       "      <td>204</td>\n",
       "      <td>Microgravity induces physiological decondition...</td>\n",
       "      <td>Skin</td>\n",
       "      <td>root:Host-associated:Human:Skin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERP019566</td>\n",
       "      <td>Longitudinal study of the diabetic skin and wo...</td>\n",
       "      <td>258</td>\n",
       "      <td>Background: Type II diabetes is a chronic heal...</td>\n",
       "      <td>Skin</td>\n",
       "      <td>root:Host-associated:Human:Skin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ERP016629</td>\n",
       "      <td>Microbiome samples derived from Buruli ulcer w...</td>\n",
       "      <td>14</td>\n",
       "      <td>Background: Buruli ulcer (BU) is an infectious...</td>\n",
       "      <td>Skin</td>\n",
       "      <td>root:Host-associated:Human:Skin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ERP021525</td>\n",
       "      <td>Micromes on salmon skin and surrounding sea water</td>\n",
       "      <td>5</td>\n",
       "      <td>Pilot study to investigate if sampled Atlantic...</td>\n",
       "      <td>Skin</td>\n",
       "      <td>root:Host-associated:Fish:Skin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRP056364</td>\n",
       "      <td>Skin microbiome in human volunteers inoculated...</td>\n",
       "      <td>191</td>\n",
       "      <td>The aim of this project was to investigate the...</td>\n",
       "      <td>Skin</td>\n",
       "      <td>root:Host-associated:Human:Skin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ERP104518</td>\n",
       "      <td>skin microbiota in infected frogs</td>\n",
       "      <td>21</td>\n",
       "      <td>skin microbiota in infected frogs</td>\n",
       "      <td>Amphibia</td>\n",
       "      <td>root:Host-associated:Amphibia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ERP104520</td>\n",
       "      <td>Skin microbiota of Scinax alcatraz</td>\n",
       "      <td>13</td>\n",
       "      <td>Skin microbiota of Scinax alcatraz</td>\n",
       "      <td>Amphibia</td>\n",
       "      <td>root:Host-associated:Amphibia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ERP104516</td>\n",
       "      <td>Variations on the diversity of amphibian skin ...</td>\n",
       "      <td>32</td>\n",
       "      <td>Variations on the diversity of amphibian skin ...</td>\n",
       "      <td>Amphibia</td>\n",
       "      <td>root:Host-associated:Amphibia</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Study ID                                         Study Name  \\\n",
       "0   ERP104068  EMG produced TPA metagenomics assembly of the ...   \n",
       "1   SRP002480  Gene-Environment Interactions at the Skin Surface   \n",
       "2   ERP018577  Human skin bacterial and fungal microbiotas an...   \n",
       "3   ERP022958  Impact of the Mk VI SkinSuit on skin microbiot...   \n",
       "4   ERP019566  Longitudinal study of the diabetic skin and wo...   \n",
       "5   ERP016629  Microbiome samples derived from Buruli ulcer w...   \n",
       "6   ERP021525  Micromes on salmon skin and surrounding sea water   \n",
       "7   SRP056364  Skin microbiome in human volunteers inoculated...   \n",
       "8   ERP104518                  skin microbiota in infected frogs   \n",
       "9   ERP104520                 Skin microbiota of Scinax alcatraz   \n",
       "10  ERP104516  Variations on the diversity of amphibian skin ...   \n",
       "\n",
       "    Number Of Samples                                     Study Abstract  \\\n",
       "0                  45  The human gut metagenome Third Party Annotatio...   \n",
       "1                2560  16S rRNA gene sequences amplified from subject...   \n",
       "2                  96  Using high-throughput 16S rDNA and ITS1 sequen...   \n",
       "3                 204  Microgravity induces physiological decondition...   \n",
       "4                 258  Background: Type II diabetes is a chronic heal...   \n",
       "5                  14  Background: Buruli ulcer (BU) is an infectious...   \n",
       "6                   5  Pilot study to investigate if sampled Atlantic...   \n",
       "7                 191  The aim of this project was to investigate the...   \n",
       "8                  21                  skin microbiota in infected frogs   \n",
       "9                  13                 Skin microbiota of Scinax alcatraz   \n",
       "10                 32  Variations on the diversity of amphibian skin ...   \n",
       "\n",
       "       Biome                          Lineage  \n",
       "0      Human       root:Host-associated:Human  \n",
       "1       Skin  root:Host-associated:Human:Skin  \n",
       "2       Skin  root:Host-associated:Human:Skin  \n",
       "3       Skin  root:Host-associated:Human:Skin  \n",
       "4       Skin  root:Host-associated:Human:Skin  \n",
       "5       Skin  root:Host-associated:Human:Skin  \n",
       "6       Skin   root:Host-associated:Fish:Skin  \n",
       "7       Skin  root:Host-associated:Human:Skin  \n",
       "8   Amphibia    root:Host-associated:Amphibia  \n",
       "9   Amphibia    root:Host-associated:Amphibia  \n",
       "10  Amphibia    root:Host-associated:Amphibia  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge table\n",
    "study_data = pd.concat([study_data, df2], axis=1)\n",
    "study_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Export study data\n",
    "study_data.to_csv('study_data.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Choose relevant study\n",
    "From the data above, we can filter which study can be used as the data for comparative microbiome analysis of human skin samples. Therefore we choose the study ID 'SRP002480' as our data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Selected study\n",
    "study = 'SRP002480'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### List samples with biomes for the given study\n",
    "\n",
    "Get study: https://www.ebi.ac.uk/metagenomics/api/latest/studies/SRP002480\n",
    "\n",
    "List samples: https://www.ebi.ac.uk/metagenomics/api/latest/studies/SRP002480/samples\n",
    "\n",
    "\n",
    "Fetch samples for the given study accession: https://www.ebi.ac.uk/metagenomics/api/latest/samples?study_accession=SRP002480\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample_from_study_SRP002480.csv\n"
     ]
    }
   ],
   "source": [
    "filename = 'sample_from_study_'+study+'.csv'\n",
    "print(filename)\n",
    "if not os.path.isfile(filename):\n",
    "    with open(filename, 'wb') as f:\n",
    "        c = pycurl.Curl()\n",
    "        c.setopt(c.URL, 'https://www.ebi.ac.uk/metagenomics/projects/'+study+'/overview/doExport')\n",
    "        c.setopt(c.WRITEDATA, f)\n",
    "        c.perform()\n",
    "        c.close()\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Script ini untuk mengekstrak data tabel dari file CSV\n",
    "df_sample_list = pd.read_csv(\"data_sample_from_studySRP002480.csv\")\n",
    "df_sample_list_refine = df_sample_list[[\"Sample ID\",\"Run ID\",\"Release version\"]]\n",
    "#df_sample_list_refine\n",
    "#df_sample_list "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'SRS451417'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sample_list_refine.loc[0, \"Sample ID\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sex</th>\n",
       "      <th>Body site</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>male</td>\n",
       "      <td>external auditory canal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>male</td>\n",
       "      <td>external auditory canal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>male</td>\n",
       "      <td>glabella</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>male</td>\n",
       "      <td>glabella</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>male</td>\n",
       "      <td>inguinal crease</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Sex                 Body site\n",
       "No                                 \n",
       "0   male        antecubital crease \n",
       "1   male        antecubital crease \n",
       "2   male                      back \n",
       "3   male                      back \n",
       "4   male   external auditory canal \n",
       "5   male   external auditory canal \n",
       "6   male           hypothenar palm \n",
       "7   male           hypothenar palm \n",
       "8   male     retroauricular crease \n",
       "9   male     retroauricular crease \n",
       "10  male                    toeweb \n",
       "11  male                    toeweb \n",
       "12  male     retroauricular crease \n",
       "13  male     retroauricular crease \n",
       "14  male                    toeweb \n",
       "15  male                    toeweb \n",
       "16  male        antecubital crease \n",
       "17  male        antecubital crease \n",
       "18  male                  glabella \n",
       "19  male                  glabella \n",
       "20  male                      back \n",
       "21  male                      back \n",
       "22  male           hypothenar palm \n",
       "23  male           hypothenar palm \n",
       "24  male           inguinal crease "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_metadata(metadata, key):\n",
    "    import html\n",
    "    for m in metadata:\n",
    "        if m['key'].lower() == key.lower():\n",
    "            value = m['value']\n",
    "            unit = html.unescape(m['unit']) if m['unit'] else \"\"\n",
    "            return \"{value} {unit}\".format(value=value, unit=unit)\n",
    "    return None\n",
    "\n",
    "df = DataFrame(columns=('Sex',\"Body site\"))\n",
    "df.index.name = 'No'\n",
    "\n",
    "#for i in range(len(df_sample_list_refine)):\n",
    "for i in range(25):\n",
    "    with Session(API_BASE) as s:\n",
    "        study = s.get('samples', df_sample_list_refine.loc[i, \"Sample ID\"]).resource\n",
    "        #print(get_metadata(study.sample_metadata, 'sex'))\n",
    "        df.loc[i] = [\n",
    "            get_metadata(study.sample_metadata, 'sex'),\n",
    "            get_metadata(study.sample_metadata, 'body site')\n",
    "        ]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Release version</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Body site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRS451417</td>\n",
       "      <td>SRR919527</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRS451417</td>\n",
       "      <td>SRR919587</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRS451418</td>\n",
       "      <td>SRR919528</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRS451418</td>\n",
       "      <td>SRR919588</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRS451419</td>\n",
       "      <td>SRR919529</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>external auditory canal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SRS451419</td>\n",
       "      <td>SRR919589</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>external auditory canal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SRS451420</td>\n",
       "      <td>SRR919530</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRS451420</td>\n",
       "      <td>SRR919590</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SRS451421</td>\n",
       "      <td>SRR919531</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SRS451421</td>\n",
       "      <td>SRR919591</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SRS451422</td>\n",
       "      <td>SRR919532</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SRS451422</td>\n",
       "      <td>SRR919592</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SRS451423</td>\n",
       "      <td>SRR919533</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SRS451423</td>\n",
       "      <td>SRR919593</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>retroauricular crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SRS451424</td>\n",
       "      <td>SRR919534</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SRS451424</td>\n",
       "      <td>SRR919594</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>toeweb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SRS451425</td>\n",
       "      <td>SRR919535</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SRS451425</td>\n",
       "      <td>SRR919595</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>antecubital crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SRS451426</td>\n",
       "      <td>SRR919537</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>glabella</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SRS451426</td>\n",
       "      <td>SRR919597</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>glabella</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SRS451427</td>\n",
       "      <td>SRR919536</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SRS451427</td>\n",
       "      <td>SRR919596</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>back</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SRS451428</td>\n",
       "      <td>SRR919538</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SRS451428</td>\n",
       "      <td>SRR919598</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>hypothenar palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SRS451429</td>\n",
       "      <td>SRR919539</td>\n",
       "      <td>2.0</td>\n",
       "      <td>male</td>\n",
       "      <td>inguinal crease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SRS451429</td>\n",
       "      <td>SRR919599</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SRS451430</td>\n",
       "      <td>SRR919540</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SRS451430</td>\n",
       "      <td>SRR919600</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SRS451431</td>\n",
       "      <td>SRR919541</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SRS451431</td>\n",
       "      <td>SRR919601</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4368</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633154</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4369</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633155</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4370</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633156</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4371</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633157</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4372</th>\n",
       "      <td>SRS732139</td>\n",
       "      <td>SRR1633158</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4373</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633147</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4374</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633148</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4375</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633149</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4376</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633150</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4377</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633151</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4378</th>\n",
       "      <td>SRS732140</td>\n",
       "      <td>SRR1633152</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4379</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633159</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4380</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633160</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4381</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633161</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4382</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633162</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4383</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633163</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4384</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633164</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4385</th>\n",
       "      <td>SRS732141</td>\n",
       "      <td>SRR1633165</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4386</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633166</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4387</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633167</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4388</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633168</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4389</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633169</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4390</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633170</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4391</th>\n",
       "      <td>SRS732142</td>\n",
       "      <td>SRR1633171</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4392</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633172</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4393</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633173</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4394</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633174</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4395</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633175</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4396</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633176</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4397</th>\n",
       "      <td>SRS732144</td>\n",
       "      <td>SRR1633177</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4398 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Sample ID      Run ID  Release version    Sex                 Body site\n",
       "0     SRS451417   SRR919527              2.0  male        antecubital crease \n",
       "1     SRS451417   SRR919587              2.0  male        antecubital crease \n",
       "2     SRS451418   SRR919528              2.0  male                      back \n",
       "3     SRS451418   SRR919588              2.0  male                      back \n",
       "4     SRS451419   SRR919529              2.0  male   external auditory canal \n",
       "5     SRS451419   SRR919589              2.0  male   external auditory canal \n",
       "6     SRS451420   SRR919530              2.0  male           hypothenar palm \n",
       "7     SRS451420   SRR919590              2.0  male           hypothenar palm \n",
       "8     SRS451421   SRR919531              2.0  male     retroauricular crease \n",
       "9     SRS451421   SRR919591              2.0  male     retroauricular crease \n",
       "10    SRS451422   SRR919532              2.0  male                    toeweb \n",
       "11    SRS451422   SRR919592              2.0  male                    toeweb \n",
       "12    SRS451423   SRR919533              2.0  male     retroauricular crease \n",
       "13    SRS451423   SRR919593              2.0  male     retroauricular crease \n",
       "14    SRS451424   SRR919534              2.0  male                    toeweb \n",
       "15    SRS451424   SRR919594              2.0  male                    toeweb \n",
       "16    SRS451425   SRR919535              2.0  male        antecubital crease \n",
       "17    SRS451425   SRR919595              2.0  male        antecubital crease \n",
       "18    SRS451426   SRR919537              2.0  male                  glabella \n",
       "19    SRS451426   SRR919597              2.0  male                  glabella \n",
       "20    SRS451427   SRR919536              2.0  male                      back \n",
       "21    SRS451427   SRR919596              2.0  male                      back \n",
       "22    SRS451428   SRR919538              2.0  male           hypothenar palm \n",
       "23    SRS451428   SRR919598              2.0  male           hypothenar palm \n",
       "24    SRS451429   SRR919539              2.0  male           inguinal crease \n",
       "25    SRS451429   SRR919599              2.0    NaN                       NaN\n",
       "26    SRS451430   SRR919540              2.0    NaN                       NaN\n",
       "27    SRS451430   SRR919600              2.0    NaN                       NaN\n",
       "28    SRS451431   SRR919541              2.0    NaN                       NaN\n",
       "29    SRS451431   SRR919601              2.0    NaN                       NaN\n",
       "...         ...         ...              ...    ...                       ...\n",
       "4368  SRS732139  SRR1633154              2.0    NaN                       NaN\n",
       "4369  SRS732139  SRR1633155              2.0    NaN                       NaN\n",
       "4370  SRS732139  SRR1633156              2.0    NaN                       NaN\n",
       "4371  SRS732139  SRR1633157              2.0    NaN                       NaN\n",
       "4372  SRS732139  SRR1633158              2.0    NaN                       NaN\n",
       "4373  SRS732140  SRR1633147              2.0    NaN                       NaN\n",
       "4374  SRS732140  SRR1633148              2.0    NaN                       NaN\n",
       "4375  SRS732140  SRR1633149              2.0    NaN                       NaN\n",
       "4376  SRS732140  SRR1633150              2.0    NaN                       NaN\n",
       "4377  SRS732140  SRR1633151              2.0    NaN                       NaN\n",
       "4378  SRS732140  SRR1633152              2.0    NaN                       NaN\n",
       "4379  SRS732141  SRR1633159              2.0    NaN                       NaN\n",
       "4380  SRS732141  SRR1633160              2.0    NaN                       NaN\n",
       "4381  SRS732141  SRR1633161              2.0    NaN                       NaN\n",
       "4382  SRS732141  SRR1633162              2.0    NaN                       NaN\n",
       "4383  SRS732141  SRR1633163              2.0    NaN                       NaN\n",
       "4384  SRS732141  SRR1633164              2.0    NaN                       NaN\n",
       "4385  SRS732141  SRR1633165              2.0    NaN                       NaN\n",
       "4386  SRS732142  SRR1633166              2.0    NaN                       NaN\n",
       "4387  SRS732142  SRR1633167              2.0    NaN                       NaN\n",
       "4388  SRS732142  SRR1633168              2.0    NaN                       NaN\n",
       "4389  SRS732142  SRR1633169              2.0    NaN                       NaN\n",
       "4390  SRS732142  SRR1633170              2.0    NaN                       NaN\n",
       "4391  SRS732142  SRR1633171              2.0    NaN                       NaN\n",
       "4392  SRS732144  SRR1633172              2.0    NaN                       NaN\n",
       "4393  SRS732144  SRR1633173              2.0    NaN                       NaN\n",
       "4394  SRS732144  SRR1633174              2.0    NaN                       NaN\n",
       "4395  SRS732144  SRR1633175              2.0    NaN                       NaN\n",
       "4396  SRS732144  SRR1633176              2.0    NaN                       NaN\n",
       "4397  SRS732144  SRR1633177              2.0    NaN                       NaN\n",
       "\n",
       "[4398 rows x 5 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result = pd.concat([df_sample_list_refine, df], axis=1)\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('List Sample'+study1+'.csv',index=True,header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sampling Data BIOM dari EBI\n",
    "https://www.ebi.ac.uk/metagenomics//projects/SRP002480/samples/SRS451457/runs/SRR919567/results/versions/2.0/taxonomy/OTU-table-HDF5-BIOM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Sample ID      Run ID\n",
      "0     SRS451417   SRR919527\n",
      "1     SRS451417   SRR919587\n",
      "2     SRS451418   SRR919528\n",
      "3     SRS451418   SRR919588\n",
      "4     SRS451419   SRR919529\n",
      "5     SRS451419   SRR919589\n",
      "6     SRS451420   SRR919530\n",
      "7     SRS451420   SRR919590\n",
      "8     SRS451421   SRR919531\n",
      "9     SRS451421   SRR919591\n",
      "10    SRS451422   SRR919532\n",
      "11    SRS451422   SRR919592\n",
      "12    SRS451423   SRR919533\n",
      "13    SRS451423   SRR919593\n",
      "14    SRS451424   SRR919534\n",
      "15    SRS451424   SRR919594\n",
      "16    SRS451425   SRR919535\n",
      "17    SRS451425   SRR919595\n",
      "18    SRS451426   SRR919537\n",
      "19    SRS451426   SRR919597\n",
      "20    SRS451427   SRR919536\n",
      "21    SRS451427   SRR919596\n",
      "22    SRS451428   SRR919538\n",
      "23    SRS451428   SRR919598\n",
      "24    SRS451429   SRR919539\n",
      "25    SRS451429   SRR919599\n",
      "26    SRS451430   SRR919540\n",
      "27    SRS451430   SRR919600\n",
      "28    SRS451431   SRR919541\n",
      "29    SRS451431   SRR919601\n",
      "...         ...         ...\n",
      "4368  SRS732139  SRR1633154\n",
      "4369  SRS732139  SRR1633155\n",
      "4370  SRS732139  SRR1633156\n",
      "4371  SRS732139  SRR1633157\n",
      "4372  SRS732139  SRR1633158\n",
      "4373  SRS732140  SRR1633147\n",
      "4374  SRS732140  SRR1633148\n",
      "4375  SRS732140  SRR1633149\n",
      "4376  SRS732140  SRR1633150\n",
      "4377  SRS732140  SRR1633151\n",
      "4378  SRS732140  SRR1633152\n",
      "4379  SRS732141  SRR1633159\n",
      "4380  SRS732141  SRR1633160\n",
      "4381  SRS732141  SRR1633161\n",
      "4382  SRS732141  SRR1633162\n",
      "4383  SRS732141  SRR1633163\n",
      "4384  SRS732141  SRR1633164\n",
      "4385  SRS732141  SRR1633165\n",
      "4386  SRS732142  SRR1633166\n",
      "4387  SRS732142  SRR1633167\n",
      "4388  SRS732142  SRR1633168\n",
      "4389  SRS732142  SRR1633169\n",
      "4390  SRS732142  SRR1633170\n",
      "4391  SRS732142  SRR1633171\n",
      "4392  SRS732144  SRR1633172\n",
      "4393  SRS732144  SRR1633173\n",
      "4394  SRS732144  SRR1633174\n",
      "4395  SRS732144  SRR1633175\n",
      "4396  SRS732144  SRR1633176\n",
      "4397  SRS732144  SRR1633177\n",
      "\n",
      "[4398 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df4 = pd.read_csv(\"skin.csv\")\n",
    "df5 = df4.loc[:,\"Sample ID\":\"Run ID\"]\n",
    "print(df5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0     SRS451417\n",
      "1     SRS451417\n",
      "2     SRS451418\n",
      "3     SRS451418\n",
      "4     SRS451419\n",
      "5     SRS451419\n",
      "6     SRS451420\n",
      "7     SRS451420\n",
      "8     SRS451421\n",
      "9     SRS451421\n",
      "10    SRS451422\n",
      "11    SRS451422\n",
      "12    SRS451423\n",
      "13    SRS451423\n",
      "14    SRS451424\n",
      "15    SRS451424\n",
      "16    SRS451425\n",
      "17    SRS451425\n",
      "18    SRS451426\n",
      "19    SRS451426\n",
      "20    SRS451427\n",
      "21    SRS451427\n",
      "22    SRS451428\n",
      "23    SRS451428\n",
      "24    SRS451429\n",
      "25    SRS451429\n",
      "Name: Sample ID, dtype: object\n"
     ]
    }
   ],
   "source": [
    "data_sampel = df5.loc[0:25,'Sample ID']\n",
    "print(data_sampel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0     SRR919527\n",
      "1     SRR919587\n",
      "2     SRR919528\n",
      "3     SRR919588\n",
      "4     SRR919529\n",
      "5     SRR919589\n",
      "6     SRR919530\n",
      "7     SRR919590\n",
      "8     SRR919531\n",
      "9     SRR919591\n",
      "10    SRR919532\n",
      "11    SRR919592\n",
      "12    SRR919533\n",
      "13    SRR919593\n",
      "14    SRR919534\n",
      "15    SRR919594\n",
      "16    SRR919535\n",
      "17    SRR919595\n",
      "18    SRR919537\n",
      "19    SRR919597\n",
      "20    SRR919536\n",
      "21    SRR919596\n",
      "22    SRR919538\n",
      "23    SRR919598\n",
      "24    SRR919539\n",
      "25    SRR919599\n",
      "Name: Run ID, dtype: object\n"
     ]
    }
   ],
   "source": [
    "data_run = df5.loc[0:25,'Run ID']\n",
    "print(data_run)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "E:\\Jupyter_Lab\\KetiakProject\\src\n"
     ]
    }
   ],
   "source": [
    "#Create ouput folder\n",
    "import os, sys\n",
    "\n",
    "cwd = os.getcwd()\n",
    "output_folder = \"\\output\" #name output folder, use \\ for directory\n",
    "\n",
    "if not os.path.isdir(cwd + output_folder):\n",
    "    os.mkdir(cwd + output_folder)\n",
    "\n",
    "new_dir = cwd + output_folder\n",
    "    \n",
    "print(cwd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRS451417_SRR919527.biom\n",
      "SRS451417_SRR919587.biom\n",
      "SRS451418_SRR919528.biom\n",
      "SRS451418_SRR919588.biom\n",
      "SRS451419_SRR919529.biom\n",
      "SRS451419_SRR919589.biom\n",
      "SRS451420_SRR919530.biom\n",
      "SRS451420_SRR919590.biom\n",
      "SRS451421_SRR919531.biom\n",
      "SRS451421_SRR919591.biom\n",
      "SRS451422_SRR919532.biom\n",
      "SRS451422_SRR919592.biom\n",
      "SRS451423_SRR919533.biom\n",
      "SRS451423_SRR919593.biom\n",
      "SRS451424_SRR919534.biom\n",
      "SRS451424_SRR919594.biom\n",
      "SRS451425_SRR919535.biom\n",
      "SRS451425_SRR919595.biom\n",
      "SRS451426_SRR919537.biom\n",
      "SRS451426_SRR919597.biom\n",
      "SRS451427_SRR919536.biom\n",
      "SRS451427_SRR919596.biom\n",
      "SRS451428_SRR919538.biom\n",
      "SRS451428_SRR919598.biom\n",
      "SRS451429_SRR919539.biom\n",
      "done\n"
     ]
    }
   ],
   "source": [
    "import pycurl\n",
    "\n",
    "#Ambil data dari EBI\n",
    "for i in range(25):\n",
    "    os.chdir(new_dir) #pindah ke folder output\n",
    "    filename = data_sampel[i] + '_' + data_run[i] + '.biom'\n",
    "    print(filename)\n",
    "    if not os.path.isfile(filename):\n",
    "        with open(filename, 'wb') as f:\n",
    "            c = pycurl.Curl()\n",
    "            c.setopt(c.URL, 'https://www.ebi.ac.uk/metagenomics//projects/SRP002480/samples/'+ data_sampel[i] + '/runs/' + data_run[i] +'/results/versions/2.0/taxonomy/OTU-table-JSON-BIOM')\n",
    "            c.setopt(c.WRITEDATA, f)\n",
    "            c.perform()\n",
    "            c.close()\n",
    "    os.chdir(cwd) #balik ke folder semula\n",
    "print('done')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### References\n",
    "---\n",
    "<a id='ref1'></a>\n",
    "1. Alex L Mitchell, Maxim Scheremetjew, Hubert Denise, Simon Potter, Aleksandra Tarkowska, Matloob Qureshi, Gustavo A Salazar, Sebastien Pesseat, Miguel A Boland, Fiona M I Hunter, Petra ten Hoopen, Blaise Alako, Clara Amid, Darren J Wilkinson, Thomas P Curtis, Guy Cochrane, Robert D Finn; EBI Metagenomics in 2017: enriching the analysis of microbial communities, from sequence reads to assemblies, Nucleic Acids Research, Volume 46, Issue D1, 4 January 2018, Pages D726–D735, https://doi.org/10.1093/nar/gkx967"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
